<!DOCTYPE html><html><head><meta charset="utf-8"><title>Demystifying NGS Quality Reports: Case Studies | Li Yao</title><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1"><meta name="theme-color" content="#3F51B5"><meta name="keywords" content="technical_docs"><meta name="description" content="Unlocking the secrets hidden in Next-Generation Sequencing (NGS) data is an exciting journey,  but it’s important to ensure that the data quality is top-notch. That’s where powerful tools like FastQC"><meta property="og:type" content="article"><meta property="og:title" content="Demystifying NGS Quality Reports: Case Studies"><meta property="og:url" content="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/index.html"><meta property="og:site_name" content="Li Yao"><meta property="og:description" content="Unlocking the secrets hidden in Next-Generation Sequencing (NGS) data is an exciting journey,  but it’s important to ensure that the data quality is top-notch. That’s where powerful tools like FastQC"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/barcode_TGGTCAC.png"><meta property="og:image" content="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/ts.png"><meta property="og:image" content="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/adapter.png"><meta property="og:image" content="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/overrepresented_sequences_CTGTAGGCACCATCA.png"><meta property="og:image" content="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/rRNA_SRR2960998.png"><meta property="article:published_time" content="2020-05-10T09:11:34.000Z"><meta property="article:modified_time" content="2025-04-23T04:03:22.936Z"><meta property="article:author" content="Li Yao"><meta property="article:tag" content="technical_docs"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/barcode_TGGTCAC.png"><link rel="alternate" type="application/atom+xml" title="Li Yao" href="/atom.xml"><link rel="shortcut icon" href="/favicon.ico"><link rel="stylesheet" href="/css/style.css?v=2.0.0"><link rel="stylesheet" href="/css/li.css?v=2.0.0"><script>window.lazyScripts=[]</script><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><meta name="generator" content="Hexo 6.2.0"></head><body><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><div id="loading" class="active"></div><aside id="menu" class="hide"><div class="inner flex-row-vertical"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="menu-off"><i class="fa-solid fa-lg fa-xmark"></i></a><div class="brand-wrap" style="background-image:url(/img/brand.jpg)"><div class="brand"> <a href="/" class="avatar waves-effect waves-circle waves-light"><img src="/img/profile.png"></a><hgroup class="introduce"><h5 class="nickname"> Li Yao</h5></hgroup></div></div><div class="scroll-wrap flex-col"><ul class="nav"><li class="waves-block waves-effect"><a href="/"><i class="icon fa-solid fa-lg fa-house-chimney"></i> Home</a></li><li class="waves-block waves-effect"><a href="/publications"><i class="icon fa-solid fa-lg fa-book"></i> Publications</a></li><li class="waves-block waves-effect"><a href="/tools"><i class="icon fa-solid fa-lg fa-rocket"></i> Tools</a></li><li class="waves-block waves-effect"><a href="/teaching"><i class="icon fa-solid fa-lg fa-person-chalkboard"></i> Teaching</a></li></ul></div></div></aside><main id="main"><div class="uni-header"><header class="top-header" id="header"><div class="flex-row"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light on" id="menu-toggle"><i class="fa-solid fa-lg fa-bars"></i></a><div class="flex-col header-title ellipsis">Demystifying NGS Quality Reports: Case Studies</div><div class="search-wrap" id="search-wrap"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="back"><i class="fa-solid fa-lg fa-chevron-left"></i></a> <input type="text" id="key" class="search-input" autocomplete="off" placeholder="Search"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="search"><i class="fa-solid fa-lg fa-magnifying-glass"></i></a></div></div></header><header class="content-header post-header"><div class="container fade-scale"><h1 class="title">Demystifying NGS Quality Reports: Case Studies</h1><h5 class="subtitle"> <time datetime="2020-05-10T09:11:34.000Z" itemprop="datePublished" class="page-time">2020-05-10</time><ul class="article-category-list"><li class="article-category-list-item"><a class="article-category-list-link" href="/posts/">posts</a></li></ul></h5></div></header></div><div class="container body-wrap"><aside class="post-widget"><nav class="post-toc-wrap post-toc-shrink" id="post-toc"><h4>Table of Content</h4><ol class="post-toc"><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Per-Base-Sequence-Content"><span class="post-toc-number">1.</span> <span class="post-toc-text">Per Base Sequence Content</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Adapter-Content"><span class="post-toc-number">2.</span> <span class="post-toc-text">Adapter Content</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Overrepresented-Sequences"><span class="post-toc-number">3.</span> <span class="post-toc-text">Overrepresented Sequences</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Possibility-1-Adapters"><span class="post-toc-number">3.1.</span> <span class="post-toc-text">Possibility 1: Adapters</span></a></li><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Possibility-2-rRNAs-or-Other-%E2%80%9CContamination%E2%80%9D"><span class="post-toc-number">3.2.</span> <span class="post-toc-text">Possibility 2: rRNAs or Other “Contamination”</span></a></li></ol></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Per-Base-Sequence-Quality"><span class="post-toc-number">4.</span> <span class="post-toc-text">Per Base Sequence Quality</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Per-Sequence-Quality-Scores"><span class="post-toc-number">5.</span> <span class="post-toc-text">Per Sequence Quality Scores</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Per-Sequence-GC-Content"><span class="post-toc-number">6.</span> <span class="post-toc-text">Per Sequence GC Content</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Per-Base-N-Content"><span class="post-toc-number">7.</span> <span class="post-toc-text">Per Base N Content</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Sequence-Length-Distribution"><span class="post-toc-number">8.</span> <span class="post-toc-text">Sequence Length Distribution</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Sequence-Duplication-Levels"><span class="post-toc-number">9.</span> <span class="post-toc-text">Sequence Duplication Levels</span></a></li></ol></nav></aside><article id="post-posts/2020-05-10-ngs-qc" class="post-article article-type-post fade" itemprop="blogPost"><div class="post-card"><h1 class="post-card-title">Demystifying NGS Quality Reports: Case Studies</h1><div class="post-meta"> <time class="post-time" title="2020-05-10 09:11:34" datetime="2020-05-10T09:11:34.000Z" itemprop="datePublished">2020-05-10</time><ul class="article-category-list" itemprop="keywords"><li class="article-category-list-item"><a class="article-category-list-link" href="/tags/technical-docs/" rel="tag">technical_docs</a></li></ul><span id="busuanzi_container_page_pv" title="Views" style="display:none"><i class="icon fa-solid fa-eye icon-pr"></i><span id="busuanzi_value_page_pv"></span></span></div><div class="post-content" id="post-content" itemprop="postContent"><p>Unlocking the secrets hidden in Next-Generation Sequencing (NGS) data is an exciting journey, but it’s important to ensure that the data quality is top-notch. That’s where powerful tools like FastQC come in! With FastQC, you can easily perform quality checks and ensure that your data is of the highest caliber. In this post, we delve into real-life cases where diagnostic plots from FastQC have helped unravel complex problems and bring clarity to our analysis. Join me on this thrilling adventure!</p><p><em>The arrangement of this post is based on the output from FastQC, but the criteria or cases mentioned here should also be applicable to similar measurements produced by other tools.</em></p><h2 id="Per-Base-Sequence-Content"><a href="#Per-Base-Sequence-Content" class="headerlink" title="Per Base Sequence Content"></a>Per Base Sequence Content</h2><p><a href="#Per-Base-Sequence-Content"><code>Per Base Sequence Content</code></a> stacks together all sequences in a fastq file and calculates the frequency of the four normal DNA bases (A&#x2F;T&#x2F;C&#x2F;G) called for each base position. In a random or unbiased library, the bases should follow a random distribution of A&#x2F;T&#x2F;C&#x2F;G, so the lines in the plot should be parallel to each other (around 25%). However, the actual distributions may have some fluctuations, depending on the overall number of bases in the genome and the capturing bias from the assay. Nevertheless, in most cases, the lines should be approximately parallel.</p><p><a href="#Per-Base-Sequence-Content"><code>Per Base Sequence Content</code></a> can be used to find:</p><ul><li>biased fragments, like:<ul><li><strong>untrimmed barcodes</strong>. For demultiplexed libraries, if there are untrimmed barcodes, then because of the fixed sequences, you would observe sharp peaks at the beginnings or ends of reads. Below is an example showing that $5^\prime$ barcodes (<em>TGGTCAC</em>) are not trimmed: <img src="/posts/demystifying-ngs-quality-reports-case-studies/barcode_TGGTCAC.png" class="" title="Per base sequence content suggests the existence of $5^prime$ barcode TGGTCAC"></li><li><strong>template switching oligo</strong>. In cases like this, you can observe characteristic trinucleotide <em>GGG</em> or <em>CCC</em> near the beginning of reads. <img src="/posts/demystifying-ngs-quality-reports-case-studies/ts.png" class="" title="Template switching oligos TATAGGG"></li></ul></li><li><a href="#Overrepresented-Sequences">overrepresented sequences</a>, like adapter dimers or rRNAs.</li></ul><p>Safelist:</p><ul><li>For libraries treated with sodium bisulfite, which will convert C to T, it’s normal to observe a low percent of Cs.</li></ul><h2 id="Adapter-Content"><a href="#Adapter-Content" class="headerlink" title="Adapter Content"></a>Adapter Content</h2><p>For libraries where a significant amount of the inserts are shorter than the sequencing length, adapters are likely to be incorporated in final reads. This is very common for libraries enriching for short&#x2F;small RNAs, like PRO-cap, PRO-seq, etc. The <a href="#Adapter-Content"><code>Adapter Content</code></a> module compares reads with commonly used adapter sequences and plots the enrichment. Adapter sequences may greatly affect on sequencing alignments, so if you see warnings in this section, you may need to trim adapters with <code>cutadapt</code>, <code>fastp</code>, or any other tool you like. In the following example, the <a href="#Adapter-Content"><code>Adapter Content</code></a> module detects the existence of <em>Nextera Transposase Sequence</em> (the risen black line).</p> <img src="/posts/demystifying-ngs-quality-reports-case-studies/adapter.png" class="" title="Untrimmed adapters"><h2 id="Overrepresented-Sequences"><a href="#Overrepresented-Sequences" class="headerlink" title="Overrepresented Sequences"></a>Overrepresented Sequences</h2><p>A sequencing library typically consists of a diverse mixture of DNA or RNA molecules, and the presence of frequently occurring specific sequences can indicate an abnormality.</p><h3 id="Possibility-1-Adapters"><a href="#Possibility-1-Adapters" class="headerlink" title="Possibility 1: Adapters"></a>Possibility 1: Adapters</h3><p>One possible cause for warnings or errors from this module is the presence of adapters. For example, people usually use a customized $3^\prime$ cloning adapter, <em>CTGTAGGCACCATCAAT</em>, to generate <a target="_blank" rel="noopener external nofollow noreferrer" href="https://en.wikipedia.org/wiki/Ribosome_profiling">Ribo-seq</a> libraries. This sequence is not included in the known-adapter list in FastQC, so the <a href="#Adapter-Content"><code>Adapter Content</code></a> module <strong>cannot</strong> detect the existence of adapters, but the <a href="#Overrepresented-Sequences"><code>Overrepresented Sequences</code></a> module catches a lot of hits. The following screenshot shows the top 12 overrepresented sequences from a Ribo-seq library (<a target="_blank" rel="noopener external nofollow noreferrer" href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR942878">SRR942878</a>), and the adapter sequence is highlighted for clarity.</p> <img src="/posts/demystifying-ngs-quality-reports-case-studies/overrepresented_sequences_CTGTAGGCACCATCA.png" class="" width="773" title="Overrepresented sequence report for sequence library SRR942878."><p>In practical applications, adapters may not be immediately visible in a table of many sequences. To identify potential adapters, I often choose a few overrepresented sequences and use the Smith-Waterman alignment algorithm to find adapter candidates. Typically, adapters appear as aligned contigs near the ends of sequences. For instance, when running the SW alignment on the first two hits in the table above, the adapter <code>CTGTAGGCACCATCAAT</code> is clearly visible:</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">CCGGCTAGCTCAGTCGGTAGAGCATGAGCTGTAGGCACCATCAATTCG</span><br><span class="line">CCAGCTAGCA-ATTGGGTGTAGC-----CTGTAGGCACCATCAATTCG</span><br></pre></td></tr></table></figure><p>You can run the Smith-Waterman algorithm <a href="bioseq.html">here</a>.</p><h3 id="Possibility-2-rRNAs-or-Other-“Contamination”"><a href="#Possibility-2-rRNAs-or-Other-“Contamination”" class="headerlink" title="Possibility 2: rRNAs or Other “Contamination”"></a>Possibility 2: rRNAs or Other “Contamination”</h3><p>What if the overrepresented sequences don’t match the typical adapter patterns? In some cases, these sequences may actually be “contamination” from ribosomal RNAs or other sources. One way to test this possibility is to run a <a target="_blank" rel="noopener external nofollow noreferrer" href="https://blast.ncbi.nlm.nih.gov/">BLAST</a> search on the sequences. For example, here are the top overrepresented sequences for <a target="_blank" rel="noopener external nofollow noreferrer" href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR2960998">another sequencing library</a>:</p> <img src="/posts/demystifying-ngs-quality-reports-case-studies/rRNA_SRR2960998.png" class="" width="977" title="Overrepresented sequence report for sequence library SRR942878."><p>When running BLAST on some of these overrepresented sequences, you can see that they match with ribosomal RNAs.</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">Chain 2, 18S rRNA</span><br><span class="line">Sequence ID: 7WTT_2 Length: 1873 Number of Matches: 1</span><br><span class="line">Range 1: 131 to 182</span><br><span class="line">Alignment statistics for match #1</span><br><span class="line">Score	Expect	Identities	Gaps	Strand</span><br><span class="line">97.1 bits(52)	1e-16	52/52(100%)	0/52(0%)	Plus/Minus</span><br><span class="line">Query  1    GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAG  52</span><br><span class="line">            ||||||||||||||||||||||||||||||||||||||||||||||||||||</span><br><span class="line">Sbjct  182  GTCGGCATGTATTAGCTCTAGAATTACCACAGTTATCCAAGTAGGAGAGGAG  131</span><br></pre></td></tr></table></figure><p>If this is the case, you can align the entire library to ribosomal RNAs first, then do a second round of alignment with reads that cannot be aligned in the first round. This ensures you are left with high-quality reads for downstream analysis.</p><h2 id="Per-Base-Sequence-Quality"><a href="#Per-Base-Sequence-Quality" class="headerlink" title="Per Base Sequence Quality"></a>Per Base Sequence Quality</h2><p>No cases yet.</p><h2 id="Per-Sequence-Quality-Scores"><a href="#Per-Sequence-Quality-Scores" class="headerlink" title="Per Sequence Quality Scores"></a>Per Sequence Quality Scores</h2><p>No cases yet.</p><h2 id="Per-Sequence-GC-Content"><a href="#Per-Sequence-GC-Content" class="headerlink" title="Per Sequence GC Content"></a>Per Sequence GC Content</h2><p>No cases yet.</p><h2 id="Per-Base-N-Content"><a href="#Per-Base-N-Content" class="headerlink" title="Per Base N Content"></a>Per Base N Content</h2><p>No cases yet.</p><h2 id="Sequence-Length-Distribution"><a href="#Sequence-Length-Distribution" class="headerlink" title="Sequence Length Distribution"></a>Sequence Length Distribution</h2><p>No cases yet.</p><h2 id="Sequence-Duplication-Levels"><a href="#Sequence-Duplication-Levels" class="headerlink" title="Sequence Duplication Levels"></a>Sequence Duplication Levels</h2><p>No cases yet.</p><p>Reference: <a target="_blank" rel="noopener external nofollow noreferrer" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">FastQC manual</a></p></div><blockquote class="post-copyright"><div class="content"><div> <span class="post-time">Last updated: <time datetime="2025-04-23T04:03:22.936Z" itemprop="dateUpdated">2025-04-23 04:03:22</time></span><br></div><div>This article is posted in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.</div><div>Permanent link to this post: <a href="https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/">https://www.yaobio.com/posts/demystifying-ngs-quality-reports-case-studies/</a></div></div><footer> <a href="https://www.yaobio.com"><img src="/img/profile.png" alt="Li Yao"> Li Yao</a></footer></blockquote><div class="post-footer"></div></div><nav class="post-nav flex-row flex-justify-between"><div class="waves-block waves-effect prev"><a href="/posts/securing-jupyter-notebooks-by-tracing-modification-histories/" id="post-prev" class="post-nav-link"><div class="tips"><i class="icon fa-solid fa-angle-left fa-lg icon-pr"></i> Prev</div><h4 class="title">Securing Jupyter Notebooks by tracing modification histories</h4></a></div><div class="waves-block waves-effect next"><a href="/posts/file-templates-for-writing-codes/" id="post-next" class="post-nav-link"><div class="tips">Next<i class="icon fa-solid fa-angle-right fa-lg icon-pl"></i></div><h4 class="title">File templates for writing codes</h4></a></div></nav><section class="comments" id="comments"><div id="disqus_thread"></div><script>var disqus_shortname="yaobio";lazyScripts.push("//"+disqus_shortname+".disqus.com/embed.js")</script><noscript>Please enable JavaScript to view the <a target="_blank" rel="noopener external nofollow noreferrer" href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript></section></article></div><footer class="footer"><div class="top"><p> <span>This page is available under the <a rel="license noopener" target="_blank" href="//creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</span></p></div><div class="bottom"><p><span>Li Yao &copy; 2019 - 2025</span> <span>Modified from indigo.</span></p></div></footer></main><div class="mask" id="mask"></div><a href="javascript:;" rel="external nofollow noreferrer" id="gotop" class="waves-effect waves-circle waves-light"><span class="fa-solid fa-lg fa-chevron-up"></span></a><script src="/js/waves.min.js?v=2.0.0"></script><script>MathJax={tex:{inlineMath:[["$","$"],["\\(","\\)"]]},svg:{fontCache:"global"}}</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><script>var BLOG={ROOT:"/",SHARE:!1,REWARD:!1}</script><script src="/js/main.js"></script><div class="search-panel" id="search-panel"><ul class="search-result" id="search-result"></ul></div><template id="search-tpl"><li class="item"><a href="{path}" class="waves-block waves-effect"><div class="title ellipsis" title="{title}">{title}</div><div class="flex-row flex-middle"><div class="tags ellipsis"> {tags}</div> <time class="flex-col time">{date}</time></div></a></li></template><script src="/js/search.min.js?v=2.0.0" async></script></body></html>