<!DOCTYPE html><html><head><meta charset="utf-8"><title>Mastering IGV: Tips and Tricks for Visualizing Sequencing Data | Li Yao</title><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1"><meta name="theme-color" content="#3F51B5"><meta name="keywords" content="technical_docs"><meta name="description" content="If a human being had actually looked at his blood, anywhere along the way,  instead of just running tests through the computer…  parasites would have jumped right out at them. “Failure to Communicate"><meta property="og:type" content="article"><meta property="og:title" content="Mastering IGV: Tips and Tricks for Visualizing Sequencing Data"><meta property="og:url" content="https://www.yaobio.com/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/index.html"><meta property="og:site_name" content="Li Yao"><meta property="og:description" content="If a human being had actually looked at his blood, anywhere along the way,  instead of just running tests through the computer…  parasites would have jumped right out at them. “Failure to Communicate"><meta property="og:locale" content="en_US"><meta property="og:image" content="https://www.yaobio.com/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/igv_load_batch_script.png"><meta property="og:image" content="https://www.yaobio.com/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/igv_without_index.png"><meta property="og:image" content="https://www.yaobio.com/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/igv_with_index.png"><meta property="article:published_time" content="2020-08-01T19:08:23.000Z"><meta property="article:modified_time" content="2025-04-23T04:03:22.944Z"><meta property="article:author" content="Li Yao"><meta property="article:tag" content="technical_docs"><meta name="twitter:card" content="summary"><meta name="twitter:image" content="https://www.yaobio.com/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/igv_load_batch_script.png"><link rel="alternate" type="application/atom+xml" title="Li Yao" href="/atom.xml"><link rel="shortcut icon" href="/favicon.ico"><link rel="stylesheet" href="/css/style.css?v=2.0.0"><link rel="stylesheet" href="/css/li.css?v=2.0.0"><script>window.lazyScripts=[]</script><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><meta name="generator" content="Hexo 6.2.0"></head><body><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><div id="loading" class="active"></div><aside id="menu" class="hide"><div class="inner flex-row-vertical"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="menu-off"><i class="fa-solid fa-lg fa-xmark"></i></a><div class="brand-wrap" style="background-image:url(/img/brand.jpg)"><div class="brand"> <a href="/" class="avatar waves-effect waves-circle waves-light"><img src="/img/profile.png"></a><hgroup class="introduce"><h5 class="nickname"> Li Yao</h5></hgroup></div></div><div class="scroll-wrap flex-col"><ul class="nav"><li class="waves-block waves-effect"><a href="/"><i class="icon fa-solid fa-lg fa-house-chimney"></i> Home</a></li><li class="waves-block waves-effect"><a href="/publications"><i class="icon fa-solid fa-lg fa-book"></i> Publications</a></li><li class="waves-block waves-effect"><a href="/tools"><i class="icon fa-solid fa-lg fa-rocket"></i> Tools</a></li><li class="waves-block waves-effect"><a href="/teaching"><i class="icon fa-solid fa-lg fa-person-chalkboard"></i> Teaching</a></li></ul></div></div></aside><main id="main"><div class="uni-header"><header class="top-header" id="header"><div class="flex-row"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light on" id="menu-toggle"><i class="fa-solid fa-lg fa-bars"></i></a><div class="flex-col header-title ellipsis">Mastering IGV: Tips and Tricks for Visualizing Sequencing Data</div><div class="search-wrap" id="search-wrap"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="back"><i class="fa-solid fa-lg fa-chevron-left"></i></a> <input type="text" id="key" class="search-input" autocomplete="off" placeholder="Search"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="search"><i class="fa-solid fa-lg fa-magnifying-glass"></i></a></div></div></header><header class="content-header post-header"><div class="container fade-scale"><h1 class="title">Mastering IGV: Tips and Tricks for Visualizing Sequencing Data</h1><h5 class="subtitle"> <time datetime="2020-08-01T19:08:23.000Z" itemprop="datePublished" class="page-time">2020-08-01</time><ul class="article-category-list"><li class="article-category-list-item"><a class="article-category-list-link" href="/posts/">posts</a></li></ul></h5></div></header></div><div class="container body-wrap"><aside class="post-widget"><nav class="post-toc-wrap post-toc-shrink" id="post-toc"><h4>Table of Content</h4><ol class="post-toc"><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Case-1-There-are-a-bunch-of-loci-that-need-to-be-checked-visually"><span class="post-toc-number">1.</span> <span class="post-toc-text">Case 1: There are a bunch of loci that need to be checked visually</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Case-2-Frequently-used-annotations-are-not-listed-in-the-default-server"><span class="post-toc-number">2.</span> <span class="post-toc-text">Case 2: Frequently used annotations are not listed in the default server</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Step-1-Copy-precompiled-data-files-from-IGV"><span class="post-toc-number">2.1.</span> <span class="post-toc-text">Step 1: Copy precompiled data files from IGV</span></a></li><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Step-2-Install-and-configure-a-web-server"><span class="post-toc-number">2.2.</span> <span class="post-toc-text">Step 2: Install and configure a web server</span></a></li><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Step-3-Save-new-annotations-and-modify-data-files"><span class="post-toc-number">2.3.</span> <span class="post-toc-text">Step 3: Save new annotations and modify data files</span></a></li><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Step-4-Change-the-data-server-setting-in-IGV"><span class="post-toc-number">2.4.</span> <span class="post-toc-text">Step 4: Change the data server setting in IGV</span></a></li></ol></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#General-tips"><span class="post-toc-number">3.</span> <span class="post-toc-text">General tips</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Always-load-bed-files-with-indices"><span class="post-toc-number">3.1.</span> <span class="post-toc-text">Always load bed files with indices</span></a></li></ol></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Conclusion"><span class="post-toc-number">4.</span> <span class="post-toc-text">Conclusion</span></a></li></ol></nav></aside><article id="post-posts/2020-08-01-IGV" class="post-article article-type-post fade" itemprop="blogPost"><div class="post-card"><h1 class="post-card-title">Mastering IGV: Tips and Tricks for Visualizing Sequencing Data</h1><div class="post-meta"> <time class="post-time" title="2020-08-01 19:08:23" datetime="2020-08-01T19:08:23.000Z" itemprop="datePublished">2020-08-01</time><ul class="article-category-list" itemprop="keywords"><li class="article-category-list-item"><a class="article-category-list-link" href="/tags/technical-docs/" rel="tag">technical_docs</a></li></ul><span id="busuanzi_container_page_pv" title="Views" style="display:none"><i class="icon fa-solid fa-eye icon-pr"></i><span id="busuanzi_value_page_pv"></span></span></div><div class="post-content" id="post-content" itemprop="postContent"><blockquote><p>If a human being had actually looked at his blood, anywhere along the way, instead of just running tests through the computer… parasites would have jumped right out at them.</p><p>“Failure to Communicate.” <em>House M.D.</em></p></blockquote><p>There is a case in the TV show House M.D., where a parasite infected the patient; Dr. House’s team runs many tests and gets no clue. In the end, Dr. House tells the team to look at the blood sample through a microscope instead of using numbers from instruments. The team does so and instantly sees the parasite. Even though the settings of the disease may not be very accurate, it leaves me with a profound impression. The same situation can also happen in bioinformatics, as we do many tests to identify targets of interest or evaluate our confidence in conclusions. However, we should be aware that these <em>tests are all based on some assumptions, and it’s always beneficial to visually check the data to have an intuitive impression that the data fit the assumptions</em>. Integrated Genome Browser (<code>IGV</code>) is a powerful tool for visualizing sequencing data. In this post, I’ll share some of my tricks for making IGV even more useful.</p><h2 id="Case-1-There-are-a-bunch-of-loci-that-need-to-be-checked-visually"><a href="#Case-1-There-are-a-bunch-of-loci-that-need-to-be-checked-visually" class="headerlink" title="Case 1: There are a bunch of loci that need to be checked visually"></a>Case 1: There are a bunch of loci that need to be checked visually</h2><p>If you need to visually check many regions of interest in IGV, you can use a batch script to automate the process. The IGV <a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/igvteam/igv/wiki/Batch-commands">batch script</a> language allows you to generate a script file that tells IGV which regions to display and where to save the output. Instead of learning the details about this minimal language yourself, you can directly use <code>bedtools</code> to create a batch script for a list of loci in a bed file. In the following example, genomic loci in <code>loci.bed</code> will first be extended 200 bp both upstream and downstream, then a batch script covering these regions will be saved to the file <code>batch.script</code>.</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">snapshots will be saved to `path_to_store_snapshots`</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">-slop indicates the number of flanking base pairs on</span> </span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">both the left and right of the interested regions to be</span> </span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">extended <span class="keyword">in</span> the captured images (0 <span class="keyword">for</span> keep them as they are)</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">I recommend <span class="built_in">set</span> the output img format to be svg or eps</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">the resolution <span class="keyword">for</span> png file is so low</span></span><br><span class="line">bedtools igv -path path_to_store_snapshots \</span><br><span class="line">	-i loci.bed -slop 200 \</span><br><span class="line">	-img svg &gt; batch.script</span><br></pre></td></tr></table></figure><p>After loading the tracks you want to see in IGV, click <code>Tools</code>&gt;<code>Run Batch Script...</code> and load the <code>batch.script</code> file, IGV will get start capturing snapshots of each locus. If the process is slow, you can split the bed file and generate batch scripts for each subset, then load them into separate instances of IGV.</p> <img src="/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/igv_load_batch_script.png" class="" title="Load batch script"><h2 id="Case-2-Frequently-used-annotations-are-not-listed-in-the-default-server"><a href="#Case-2-Frequently-used-annotations-are-not-listed-in-the-default-server" class="headerlink" title="Case 2: Frequently used annotations are not listed in the default server"></a>Case 2: Frequently used annotations are not listed in the default server</h2><p>The IGV team maintains a fabulous web server with some commonly used annotations (like gene annotations from the GENCODE project) or datasets (like ChIP-seq alignments from the ENCODE project); by simply selecting the annotations of interest from <code>File</code>&gt;<code>Load from Server...</code>, you can load them to your current session. One small pitfall with this function is that the annotations or datasets are not always up-to-date; for some frequently used files (or customized files), you may want them listed there. In this case, you should consider setting up your data server for IGV.</p><h3 id="Step-1-Copy-precompiled-data-files-from-IGV"><a href="#Step-1-Copy-precompiled-data-files-from-IGV" class="headerlink" title="Step 1: Copy precompiled data files from IGV"></a>Step 1: Copy precompiled data files from IGV</h3><p>You can get a copy of all genome files that IGV is currently using from their GitHub repo:</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">git clone https://github.com/igvteam/igv.git</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">igv team removed genome files <span class="keyword">in</span> commit 218f873</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">so we need to check out from one commit before the deletion,</span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash"><span class="built_in">which</span> refers to commit beb4f48</span></span><br><span class="line">git checkout beb4f48e04</span><br></pre></td></tr></table></figure><p>After checkout, you can copy the entire <code>igv/genomes</code> folder to a new place (assuming it’s <code>/nas1/references</code>) and set up your data server.</p><h3 id="Step-2-Install-and-configure-a-web-server"><a href="#Step-2-Install-and-configure-a-web-server" class="headerlink" title="Step 2: Install and configure a web server"></a>Step 2: Install and configure a web server</h3><p>If you’ve already had a web server, then you can move to step 3. For Mac users, you can install Nginx with <a target="_blank" rel="noopener external nofollow noreferrer" href="https://brew.sh/">Homebrew</a>:</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">brew install nginx</span><br></pre></td></tr></table></figure><p>By default, the configuration file for Nginx (installed by Homebrew) is located at <code>/usr/local/etc/nginx/nginx.conf</code>. In the <code>http</code> section, add a new server configuration as follows:</p><figure class="highlight nginx"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="section">server</span> &#123;</span><br><span class="line">  <span class="attribute">listen</span> <span class="number">80</span>;</span><br><span class="line">  <span class="attribute">listen</span> [::]:<span class="number">80</span>;</span><br><span class="line">  <span class="comment"># if you have a domain, replace `ref.yaobio.com` with your own domain</span></span><br><span class="line">  <span class="comment"># if you don&#x27;t have one, and only want to access the web server locally</span></span><br><span class="line">  <span class="comment"># you can replace it with localhost</span></span><br><span class="line">  <span class="attribute">server_name</span> ref.yaobio.com;</span><br><span class="line">  </span><br><span class="line">  <span class="section">location</span> / &#123;</span><br><span class="line">    <span class="comment"># replace this path with where you put the data files from IGV</span></span><br><span class="line">    <span class="attribute">root</span>   /nas1/references;</span><br><span class="line">    <span class="attribute">index</span>  index.html index.htm;</span><br><span class="line">  &#125;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure><p>Reload the configurations to make changes effective:</p><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">nginx reload</span><br></pre></td></tr></table></figure><p>Create a new directory (<code>annotations</code>) in <code>/nas1/references</code>. Now the structure of this folder is something like</p><ul><li>references<ul><li>db<ul><li>1kg_ref</li><li>…</li><li>hg19</li><li>hg38</li><li>mm10</li><li>…</li></ul></li><li>sizes<ul><li>1kg_ref.chrom.sizes</li><li>…</li><li>hg19.chrom.sizes</li><li>hg38.chrom.sizes</li><li>mm10.chrom.sizes</li><li>…</li></ul></li><li>annotations</li><li>genomes.tab</li><li>genomes.txt</li></ul></li></ul><h3 id="Step-3-Save-new-annotations-and-modify-data-files"><a href="#Step-3-Save-new-annotations-and-modify-data-files" class="headerlink" title="Step 3: Save new annotations and modify data files"></a>Step 3: Save new annotations and modify data files</h3><p>Let’s assume you have a new annotation file processed (e.g., processed GENCODE v35 for hg38 with the pipeline we mentioned in the previous <a href="/posts/collection-of-commonly-used-references-in-bioinformatic-analysis/" title="post">post</a>); now, you can move the file to <code>/nas1/references/annotations</code>. Then you need to modify the default genome and data registry:</p><ol><li><p>Change the content of <code>db/hg38/hg38_dataServerRegistry.txt</code> from <code>https://s3.amazonaws.com/igv.org.genomes/hg38/hg38_annotations.xml</code> to <code>https://ref.yaobio.com/db/hg38/hg38_annotations.xml</code></p></li><li><p>Add the following item to <code>db/hg38/hg38_annotations.xml</code>:</p><figure class="highlight xml"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"><span class="tag">&lt;<span class="name">Resource</span> <span class="attr">name</span>=<span class="string">&quot;Gencode V35&quot;</span> <span class="attr">path</span>=<span class="string">&quot;http://ref.yaobio.com/annotations/gencode.v35.annotation.sorted.gtf.gz&quot;</span> <span class="attr">index</span>=<span class="string">&quot;http://ref.yaobio.com/annotations/gencode.v35.annotation.sorted.gtf.gz.tbi&quot;</span> <span class="attr">hyperlink</span>=<span class="string">&quot;http://www.gencodegenes.org/&quot;</span>/&gt;</span></span><br></pre></td></tr></table></figure></li></ol><h3 id="Step-4-Change-the-data-server-setting-in-IGV"><a href="#Step-4-Change-the-data-server-setting-in-IGV" class="headerlink" title="Step 4: Change the data server setting in IGV"></a>Step 4: Change the data server setting in IGV</h3><p>Now in IGV, click <code>View</code>&gt;<code>Preferences</code>&gt;<code>Advanced</code>, and replace the previous value in <code>Data registry url</code> with <code>http://ref.yaobio.com/db/$$/$$_dataServerRegistry.txt</code>. Finally, save the changes, and restart IGV; you should be able to see and load newly added annotations into IGV.</p><h2 id="General-tips"><a href="#General-tips" class="headerlink" title="General tips"></a>General tips</h2><h3 id="Always-load-bed-files-with-indices"><a href="#Always-load-bed-files-with-indices" class="headerlink" title="Always load bed files with indices"></a>Always load bed files with indices</h3><p>Always create indices for bed files before loading them into IGV. Otherwise, IGV will read every interval into memory and generate indexes on the fly, consuming excessive memory and time. You can use <code>tabix</code> to generate an index for interval files before loading them into IGV. This practice can greatly reduce memory usage and computation time. Let’s say we have an interval bed file <code>test_file_1.bed.gz</code>, it has 18M records; after loading this file into IGV without index, IGV takes more than <strong>25GB</strong> of memory!</p> <img src="/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/igv_without_index.png" class="" title="Without index, IGV takes significant amount of memory"><p>But if you use <code>tabix test_file_1.bed.gz</code> to generate the index first, and then feed IGV with the same file, it only takes <strong>2GB</strong>!</p> <img src="/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/igv_with_index.png" class="" title="With index, IGV takes relatively less memory"><h2 id="Conclusion"><a href="#Conclusion" class="headerlink" title="Conclusion"></a>Conclusion</h2><p>IGV is a valuable tool for visualizing sequencing data, and these tips and tricks can help you make the most of its capabilities. By using batch scripting, setting up a personal data server, and optimizing bed file loading with indices, you can streamline your bioinformatics workflows and gain more insights from your data.</p></div><blockquote class="post-copyright"><div class="content"><div> <span class="post-time">Last updated: <time datetime="2025-04-23T04:03:22.944Z" itemprop="dateUpdated">2025-04-23 04:03:22</time></span><br></div><div>This article is posted in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.</div><div>Permanent link to this post: <a href="https://www.yaobio.com/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/">https://www.yaobio.com/posts/mastering-igv-tips-and-tricks-for-visualizing-sequencing-data/</a></div></div><footer> <a href="https://www.yaobio.com"><img src="/img/profile.png" alt="Li Yao"> Li Yao</a></footer></blockquote><div class="post-footer"></div></div><nav class="post-nav flex-row flex-justify-between"><div class="waves-block waves-effect prev"><a href="/posts/case-study-resolving-the-mystery-of-a-full-linux-root-disk/" id="post-prev" class="post-nav-link"><div class="tips"><i class="icon fa-solid fa-angle-left fa-lg icon-pr"></i> Prev</div><h4 class="title">Case Study: Resolving the Mystery of a Full Linux Root Disk</h4></a></div><div class="waves-block waves-effect next"><a href="/posts/setting-up-a-handy-research-environment-on-linux-servers/" id="post-next" class="post-nav-link"><div class="tips">Next<i class="icon fa-solid fa-angle-right fa-lg icon-pl"></i></div><h4 class="title">Setting up a handy research environment on Linux servers</h4></a></div></nav><section class="comments" id="comments"><div id="disqus_thread"></div><script>var disqus_shortname="yaobio";lazyScripts.push("//"+disqus_shortname+".disqus.com/embed.js")</script><noscript>Please enable JavaScript to view the <a target="_blank" rel="noopener external nofollow noreferrer" href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript></section></article></div><footer class="footer"><div class="top"><p> <span>This page is available under the <a rel="license noopener" target="_blank" href="//creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</span></p></div><div class="bottom"><p><span>Li Yao &copy; 2019 - 2025</span> <span>Modified from indigo.</span></p></div></footer></main><div class="mask" id="mask"></div><a href="javascript:;" rel="external nofollow noreferrer" id="gotop" class="waves-effect waves-circle waves-light"><span class="fa-solid fa-lg fa-chevron-up"></span></a><script src="/js/waves.min.js?v=2.0.0"></script><script>MathJax={tex:{inlineMath:[["$","$"],["\\(","\\)"]]},svg:{fontCache:"global"}}</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><script>var BLOG={ROOT:"/",SHARE:!1,REWARD:!1}</script><script src="/js/main.js"></script><div class="search-panel" id="search-panel"><ul class="search-result" id="search-result"></ul></div><template id="search-tpl"><li class="item"><a href="{path}" class="waves-block waves-effect"><div class="title ellipsis" title="{title}">{title}</div><div class="flex-row flex-middle"><div class="tags ellipsis"> {tags}</div> <time class="flex-col time">{date}</time></div></a></li></template><script src="/js/search.min.js?v=2.0.0" async></script></body></html>