<!DOCTYPE html><html><head><meta charset="utf-8"><title>Publication | Li Yao</title><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1"><meta name="theme-color" content="#3F51B5"><meta name="keywords" content=""><meta property="og:type" content="website"><meta property="og:title" content="Li Yao"><meta property="og:url" content="https://www.yaobio.com/publication/page/2/index.html"><meta property="og:site_name" content="Li Yao"><meta property="og:locale" content="en_US"><meta property="article:author" content="Li Yao"><meta name="twitter:card" content="summary"><link rel="alternate" type="application/atom+xml" title="Li Yao" href="/atom.xml"><link rel="shortcut icon" href="/favicon.ico"><link rel="stylesheet" href="/css/style.css?v=2.0.0"><link rel="stylesheet" href="/css/li.css?v=2.0.0"><script>window.lazyScripts=[]</script><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><meta name="generator" content="Hexo 6.2.0"></head><body><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><div id="loading" class="active"></div><aside id="menu"><div class="inner flex-row-vertical"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="menu-off"><i class="fa-solid fa-lg fa-xmark"></i></a><div class="brand-wrap" style="background-image:url(/img/brand.jpg)"><div class="brand"> <a href="/" class="avatar waves-effect waves-circle waves-light"><img src="/img/profile.png"></a><hgroup class="introduce"><h5 class="nickname"> Li Yao</h5></hgroup></div></div><div class="scroll-wrap flex-col"><ul class="nav"><li class="waves-block waves-effect"><a href="/"><i class="icon fa-solid fa-lg fa-house-chimney"></i> Home</a></li><li class="waves-block waves-effect"><a href="/publications"><i class="icon fa-solid fa-lg fa-book"></i> Publications</a></li><li class="waves-block waves-effect"><a href="/tools"><i class="icon fa-solid fa-lg fa-rocket"></i> Tools</a></li><li class="waves-block waves-effect"><a href="/teaching"><i class="icon fa-solid fa-lg fa-person-chalkboard"></i> Teaching</a></li></ul></div></div></aside><main id="main"><div class="uni-header"><header class="top-header" id="header"><div class="flex-row"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light on" id="menu-toggle"><i class="fa-solid fa-lg fa-bars"></i></a><div class="flex-col header-title ellipsis">Publication</div><div class="search-wrap" id="search-wrap"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="back"><i class="fa-solid fa-lg fa-chevron-left"></i></a> <input type="text" id="key" class="search-input" autocomplete="off" placeholder="Search"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="search"><i class="fa-solid fa-lg fa-magnifying-glass"></i></a></div></div></header><header class="content-header index-header"><div class="container fade-scale"><h1 class="title">Publication</h1><h5 class="subtitle"></h5></div></header></div><div class="container body-wrap"><ul class="post-list"><li class="post-list-item fade"><article id="post-publications/2019-interaction-variant-hpop" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2019-08-06 14:37:20" datetime="2019-08-06T14:37:20.000Z" itemprop="datePublished">2019-08-06</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/extensive-disruption-of-protein-interactions-by-genetic-variants-across-the-allele-frequency-spectrum-in-human-populations/">Extensive disruption of protein interactions by genetic variants across the allele frequency spectrum in human populations</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Each human genome carries tens of thousands of coding variants. The extent to which this variation is functional and the mechanisms by which they exert their influence remains largely unexplored. To address this gap, we leverage the ExAC database of 60,706 human exomes to investigate experimentally the impact of 2009 missense single nucleotide variants (SNVs) across 2185 protein-protein interactions, generating interaction profiles for 4797 SNV-interaction pairs, of which 421 SNVs segregate at &gt; 1% allele frequency in human populations. We find that interaction-disruptive SNVs are prevalent at both rare and common allele frequencies. Furthermore, these results suggest that 10.5% of missense variants carried per individual are disruptive, a higher proportion than previously reported; this indicates that each individual’s genetic makeup may be significantly more complex than expected. Finally, we demonstrate that candidate disease-associated mutations can be identified through shared interaction perturbations between variants of interest and known disease mutations. <a href="/publication/extensive-disruption-of-protein-interactions-by-genetic-variants-across-the-allele-frequency-spectrum-in-human-populations/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2018-biosyntax" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2018-08-14 14:34:43" datetime="2018-08-14T14:34:43.000Z" itemprop="datePublished">2018-08-14</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/biosyntax-syntax-highlighting-for-computational-biology/">bioSyntax: syntax highlighting for computational biology</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Background: Computational biology requires the reading and comprehension of biological data files. Plain-text formats such as SAM, VCF, GTF, PDB and FASTA, often contain critical information which is obfuscated by the data structure complexity. Results: bioSyntax is a freely available suite of biological syntax highlighting packages for vim, gedit, Sublime, VSCode, and less. bioSyntax improves the legibility of low-level biological data in the bioinformatics workspace. Conclusion: bioSyntax supports computational scientists in parsing and comprehending their data efficiently and thus can accelerate research output. <a href="/publication/biosyntax-syntax-highlighting-for-computational-biology/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2017-rna-editing-plus" class="article-card article-type-post" itemprop="blogPost" style="padding-bottom:16px"><div class="post-flex"><div class="post-thumbnail"> <img src="/assets/covers/rep_bib_2019_v20_i1.png"></div><div class="post-wrap"><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/large-scale-prediction-of-adar-mediated-effective-human-a-to-i-rna-editing/">Large-scale prediction of ADAR-mediated effective human A-to-I RNA editing</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Adenosine-to-inosine (A-to-I) editing by adenosine deaminase acting on the RNA (ADAR) proteins is one of the most frequent modifications during post- and co-transcription. To facilitate the assignment of biological functions to specific editing sites, we designed an automatic online platform to annotate A-to-I RNA editing sites in pre-mRNA splicing signals, microRNAs (miRNAs) and miRNA target untranslated regions ($3^\prime$ UTRs) from human (Homo sapiens) high-throughput sequencing data and predict their effects based on large-scale bioinformatic analysis. After analysing plenty of previously reported RNA editing events and human normal tissues RNA high-seq data, &gt;60000 potentially effective RNA editing events on functional genes were found. The RNA Editing Plus platform is available for free at https://www.rnaeditplus.org/, and we believe our platform governing multiple optimized methods will improve further studies of A-to-I-induced editing post-transcriptional regulation. <a href="/publication/large-scale-prediction-of-adar-mediated-effective-human-a-to-i-rna-editing/" class="post-more waves-effect waves-button">Continue reading...</a></div></div></div></article></li><li class="post-list-item fade"><article id="post-publications/2017-bioqueue" class="article-card article-type-post" itemprop="blogPost" style="padding-bottom:16px"><div class="post-flex"><div class="post-thumbnail"> <img src="/assets/covers/bioq_bioinfo_2017_v33_i20.png"></div><div class="post-wrap"><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/bioqueue-a-novel-pipeline-framework-to-accelerate-bioinformatics-analysis/">BioQueue: a novel pipeline framework to accelerate bioinformatics analysis</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Motivation: With the rapid development of Next-Generation Sequencing, a large amount of data is now available for bioinformatics research. Meanwhile, the presence of many pipeline frameworks makes it possible to analyse these data. However, these tools concentrate mainly on their syntax and design paradigms, and dispatch jobs based on users’ experience about the resources needed by the execution of a certain step in a protocol. As a result, it is difficult for these tools to maximize the potential of computing resources, and avoid errors caused by overload, such as memory overflow. Results: Here, we have developed BioQueue, a web-based framework that contains a checkpoint before each step to automatically estimate the system resources (CPU, memory and disk) needed by the step and then dispatch jobs accordingly. BioQueue possesses a shell command-like syntax instead of implementing a new script language, which means most biologists without computer programming background can access the efficient queue system with ease. <a href="/publication/bioqueue-a-novel-pipeline-framework-to-accelerate-bioinformatics-analysis/" class="post-more waves-effect waves-button">Continue reading...</a></div></div></div></article></li><li class="post-list-item fade"><article id="post-publications/2016-circulating-mirna" class="article-card article-type-post" itemprop="blogPost" style="padding-bottom:16px"><div class="post-flex"><div class="post-thumbnail"> <img src="/assets/covers/dna.2017.36.issue-2.cover.jpg"></div><div class="post-wrap"><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/circulating-micrornas-promising-biomarkers-involved-in-several-cancers-and-other-diseases/">Circulating microRNAs: Promising Biomarkers Involved in Several Cancers and Other Diseases</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Recently, many studies indicated that microRNAs (miRNAs) stably existed in various body fluids, including serum, plasma, saliva, and urine. Such miRNAs that exist in mammalian body fluids are known as circulating miRNAs, and they can transmit signals between cells and regulate intracellular gene expression. Currently, we barely understand the characteristics, sources, secretion, uptake, and functions of newly generated miRNAs. Particularly, it has been shown that certain types of circulating miRNAs can provide effective clinical data, suggesting their roles as novel biomarkers for the early detection of diseases such as cancers, cardiovascular diseases, and diabetes. Therefore, miRNAs have attracted much attention in academia for their promising applications in fundamental research and clinical diagnosis. This review summarizes some of the functional studies that have been conducted as well as the promising applications of circulating miRNAs, and we hope it will benefit other researchers in this field. <a href="/publication/circulating-micrornas-promising-biomarkers-involved-in-several-cancers-and-other-diseases/" class="post-more waves-effect waves-button">Continue reading...</a></div></div></div></article></li></ul><nav id="page-nav"><div class="inner"> <a class="extend prev" rel="prev" href="/publication/">Previous</a><a class="page-number" href="/publication/">1</a><span class="page-number current">2</span></div></nav></div><footer class="footer"><div class="top"><p> <span>This page is available under the <a rel="license noopener" target="_blank" href="//creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</span></p></div><div class="bottom"><p><span>Li Yao &copy; 2019 - 2025</span> <span>Modified from indigo.</span></p></div></footer></main><div class="mask" id="mask"></div><a href="javascript:;" rel="external nofollow noreferrer" id="gotop" class="waves-effect waves-circle waves-light"><span class="fa-solid fa-lg fa-chevron-up"></span></a><script src="/js/waves.min.js?v=2.0.0"></script><script>MathJax={tex:{inlineMath:[["$","$"],["\\(","\\)"]]},svg:{fontCache:"global"}}</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><script>var BLOG={ROOT:"/",SHARE:!1,REWARD:!1}</script><script src="/js/main.js"></script><div class="search-panel" id="search-panel"><ul class="search-result" id="search-result"></ul></div><template id="search-tpl"><li class="item"><a href="{path}" class="waves-block waves-effect"><div class="title ellipsis" title="{title}">{title}</div><div class="flex-row flex-middle"><div class="tags ellipsis"> {tags}</div> <time class="flex-col time">{date}</time></div></a></li></template><script src="/js/search.min.js?v=2.0.0" async></script></body></html>