<!DOCTYPE html><html><head><meta charset="utf-8"><title>Collection of commonly used references in bioinformatic analysis | Li Yao</title><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1"><meta name="theme-color" content="#3F51B5"><meta name="keywords" content="resources"><meta name="description" content="HumanCurrently, there are two widely used releases GRCh38 (hg38) and GRCh37 (hg19). GRCh38 (hg38)Sequences File name: GRCh38_no_alt_analysis_set_GCA_000001405.15.fa.gz (MD5 checksum: a08035b6a6e31780e"><meta property="og:type" content="article"><meta property="og:title" content="Collection of commonly used references in bioinformatic analysis"><meta property="og:url" content="https://www.yaobio.com/posts/collection-of-commonly-used-references-in-bioinformatic-analysis/index.html"><meta property="og:site_name" content="Li Yao"><meta property="og:description" content="HumanCurrently, there are two widely used releases GRCh38 (hg38) and GRCh37 (hg19). GRCh38 (hg38)Sequences File name: GRCh38_no_alt_analysis_set_GCA_000001405.15.fa.gz (MD5 checksum: a08035b6a6e31780e"><meta property="og:locale" content="en_US"><meta property="article:published_time" content="2020-04-22T17:41:22.000Z"><meta property="article:modified_time" content="2025-04-23T04:03:22.945Z"><meta property="article:author" content="Li Yao"><meta property="article:tag" content="resources"><meta name="twitter:card" content="summary"><link rel="alternate" type="application/atom+xml" title="Li Yao" href="/atom.xml"><link rel="shortcut icon" href="/favicon.ico"><link rel="stylesheet" href="/css/style.css?v=2.0.0"><link rel="stylesheet" href="/css/li.css?v=2.0.0"><script>window.lazyScripts=[]</script><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><meta name="generator" content="Hexo 6.2.0"></head><body><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><div id="loading" class="active"></div><aside id="menu" class="hide"><div class="inner flex-row-vertical"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="menu-off"><i class="fa-solid fa-lg fa-xmark"></i></a><div class="brand-wrap" style="background-image:url(/img/brand.jpg)"><div class="brand"> <a href="/" class="avatar waves-effect waves-circle waves-light"><img src="/img/profile.png"></a><hgroup class="introduce"><h5 class="nickname"> Li Yao</h5></hgroup></div></div><div class="scroll-wrap flex-col"><ul class="nav"><li class="waves-block waves-effect"><a href="/"><i class="icon fa-solid fa-lg fa-house-chimney"></i> Home</a></li><li class="waves-block waves-effect"><a href="/publications"><i class="icon fa-solid fa-lg fa-book"></i> Publications</a></li><li class="waves-block waves-effect"><a href="/tools"><i class="icon fa-solid fa-lg fa-rocket"></i> Tools</a></li><li class="waves-block waves-effect"><a href="/teaching"><i class="icon fa-solid fa-lg fa-person-chalkboard"></i> Teaching</a></li></ul></div></div></aside><main id="main"><div class="uni-header"><header class="top-header" id="header"><div class="flex-row"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light on" id="menu-toggle"><i class="fa-solid fa-lg fa-bars"></i></a><div class="flex-col header-title ellipsis">Collection of commonly used references in bioinformatic analysis</div><div class="search-wrap" id="search-wrap"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="back"><i class="fa-solid fa-lg fa-chevron-left"></i></a> <input type="text" id="key" class="search-input" autocomplete="off" placeholder="Search"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="search"><i class="fa-solid fa-lg fa-magnifying-glass"></i></a></div></div></header><header class="content-header post-header"><div class="container fade-scale"><h1 class="title">Collection of commonly used references in bioinformatic analysis</h1><h5 class="subtitle"> <time datetime="2020-04-22T17:41:22.000Z" itemprop="datePublished" class="page-time">2020-04-22</time><ul class="article-category-list"><li class="article-category-list-item"><a class="article-category-list-link" href="/posts/">posts</a></li></ul></h5></div></header></div><div class="container body-wrap"><aside class="post-widget"><nav class="post-toc-wrap post-toc-shrink" id="post-toc"><h4>Table of Content</h4><ol class="post-toc"><li class="post-toc-item post-toc-level-1"><a class="post-toc-link" href="#Human"><span class="post-toc-number">1.</span> <span class="post-toc-text">Human</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#GRCh38-hg38"><span class="post-toc-number">1.1.</span> <span class="post-toc-text">GRCh38 (hg38)</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Sequences"><span class="post-toc-number">1.1.1.</span> <span class="post-toc-text">Sequences</span></a></li><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Annotations"><span class="post-toc-number">1.1.2.</span> <span class="post-toc-text">Annotations</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#Gene-annotations"><span class="post-toc-number">1.1.2.1.</span> <span class="post-toc-text">Gene annotations</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-5"><a class="post-toc-link" href="#GENCODE"><span class="post-toc-number">1.1.2.1.1.</span> <span class="post-toc-text">GENCODE</span></a></li><li class="post-toc-item post-toc-level-5"><a class="post-toc-link" href="#RefGene"><span class="post-toc-number">1.1.2.1.2.</span> <span class="post-toc-text">RefGene</span></a></li></ol></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#Other-annotations"><span class="post-toc-number">1.1.2.2.</span> <span class="post-toc-text">Other annotations</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-5"><a class="post-toc-link" href="#Repeat-Masker"><span class="post-toc-number">1.1.2.2.1.</span> <span class="post-toc-text">Repeat Masker</span></a></li></ol></li></ol></li></ol></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Generic"><span class="post-toc-number">1.2.</span> <span class="post-toc-text">Generic</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Sequences-1"><span class="post-toc-number">1.2.1.</span> <span class="post-toc-text">Sequences</span></a></li><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#Annotations-1"><span class="post-toc-number">1.2.2.</span> <span class="post-toc-text">Annotations</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#Motif-databases-MEME"><span class="post-toc-number">1.2.2.1.</span> <span class="post-toc-text">Motif databases (MEME)</span></a></li></ol></li></ol></li></ol></li><li class="post-toc-item post-toc-level-1"><a class="post-toc-link" href="#Note"><span class="post-toc-number">2.</span> <span class="post-toc-text">Note</span></a></li></ol></nav></aside><article id="post-posts/references" class="post-article article-type-post fade" itemprop="blogPost"><div class="post-card"><h1 class="post-card-title">Collection of commonly used references in bioinformatic analysis</h1><div class="post-meta"> <time class="post-time" title="2020-04-22 17:41:22" datetime="2020-04-22T17:41:22.000Z" itemprop="datePublished">2020-04-22</time><ul class="article-category-list" itemprop="keywords"><li class="article-category-list-item"><a class="article-category-list-link" href="/tags/resources/" rel="tag">resources</a></li></ul><span id="busuanzi_container_page_pv" title="Views" style="display:none"><i class="icon fa-solid fa-eye icon-pr"></i><span id="busuanzi_value_page_pv"></span></span></div><div class="post-content" id="post-content" itemprop="postContent"><h1 id="Human"><a href="#Human" class="headerlink" title="Human"></a>Human</h1><p>Currently, there are two widely used releases GRCh38 (hg38) and GRCh37 (hg19).</p><h2 id="GRCh38-hg38"><a href="#GRCh38-hg38" class="headerlink" title="GRCh38 (hg38)"></a>GRCh38 (hg38)</h2><h3 id="Sequences"><a href="#Sequences" class="headerlink" title="Sequences"></a>Sequences</h3><ul><li>File name: <code>GRCh38_no_alt_analysis_set_GCA_000001405.15.fa.gz</code> (MD5 checksum: <code>a08035b6a6e31780e96a34008ff21bd6</code>)</li><li>Local path: <em>&#x2F;References&#x2F;Sequences&#x2F;human&#x2F;hg38&#x2F;GRCh38_no_alt_analysis_set_GCA_000001405.15.fa.gz</em></li><li>Remote backup: <a target="_blank" rel="noopener external nofollow noreferrer" href="https://osf.io/qyznb/">OSF</a></li><li>Description: This file contains sequences for the following:<ol><li>chromosomes from the GRCh38 Primary Assembly (PA);</li><li>mitochondrial genome from the GRCh38 non-nuclear assembly;</li><li>unlocalized scaffolds from PA;</li><li>unplaced scaffolds from PA;</li><li>Epstein-Barr virus (EBV) sequence.</li></ol></li><li>Recipe:<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wget https://www.encodeproject.org/files/GRCh38_no_alt_analysis_set_GCA_000001405.15/@@download/GRCh38_no_alt_analysis_set_GCA_000001405.15.fasta.gz;</span><br></pre></td></tr></table></figure></li></ul><h3 id="Annotations"><a href="#Annotations" class="headerlink" title="Annotations"></a>Annotations</h3><h4 id="Gene-annotations"><a href="#Gene-annotations" class="headerlink" title="Gene annotations"></a>Gene annotations</h4><p>There are three major releases of gene annotations for <em>Homo sapiens</em>:</p><ul><li>GENCODE&#x2F;Ensembl annotation: The GENCODE annotation is made from Ensembl annotation, so gene annotations are the same in both releases. The only exception is that the genes which are common to the human chromosome X and Y PAR regions can be found twice in the GENCODE GTF, while they are shown only for chromosome X in the Ensembl file. Gene &#x2F; transcripts IDs are the same in both releases except for annotations is the PAR regions. Comparing to other annotations, GENCODE annotation provides higher coverage among non-coding regions.</li><li>RefSeq Gene (RefGene): Annotations for well-characterized genes (mostly protein-coding genes). Projects like Gene Ontology, KEGG and MSigDB (Molecular Signatures Database, gene sets for GSEA) use this annotation as gene identifiers. So RefGene maybe the preferred annotation if you want to do enrichment analysis with GO&#x2F;KEGG&#x2F;GSEA.</li><li>UCSC Known genes: Automatically generated annotations (based on protein sequences from Swiss-Prot), mostly for protein-coding genes.</li></ul><h5 id="GENCODE"><a href="#GENCODE" class="headerlink" title="GENCODE"></a>GENCODE</h5><ul><li><p>File name: <code>gencode.v24.annotation.gtf.gz</code> (MD5 checksum: <code>17395005bb4471605db62042b992893e</code>)</p></li><li><p>Local path: <em>&#x2F;References&#x2F;Annotations&#x2F;human&#x2F;hg38&#x2F;gencode.v24.annotation.gtf.gz</em></p></li><li><p>Remote backup: <a target="_blank" rel="noopener external nofollow noreferrer" href="https://osf.io/v4rn9/">OSF</a></p></li><li><p>Description: GENCODE comprehensive annotation release 24. Downloaded from GENCODEâ€™s website.</p></li><li><p>Recipe:</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_24/gencode.v24.annotation.gtf.gz</span><br></pre></td></tr></table></figure></li><li><p>File name: <code>gencode.v24.segmented.tssup1kb.bed.gz</code> (MD5 checksum: <code>972a57431c6209667d5aac41bbb01ebd</code>)</p></li><li><p>Local path: <em>&#x2F;References&#x2F;Annotations&#x2F;human&#x2F;hg38&#x2F;gencode.v24.segmented.tss*<em>up1kb</em></em>.bed.gz*</p></li><li><p>Remote backup: <a target="_blank" rel="noopener external nofollow noreferrer" href="https://osf.io/ptde7/">OSF</a></p></li><li><p>Description: Genomic segmentations (promoter, 5_UTR, exon, intron, 3_UTR, and intergenic region) based on GENCODE v24, promoters were defined as upstream 1kb of TSSs (transcripts).</p></li><li><p>Recipe:</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">promoters <span class="keyword">for</span> protein-coding genes</span></span><br><span class="line">zcat gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; $3==&quot;transcript&quot; &#123;print $1,$4-1,$5,$18,&quot;promoter&quot;,$7,$14&#125;&#x27; | tr -d &#x27;&quot;;&#x27; | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;if ($7==&quot;protein_coding&quot;)&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;102,194,165&quot;&#125;&#125;&#x27; | \</span><br><span class="line">    bedtools flank -i - -g hg38.genome -l 1000 -r 0 -s &gt; promoters_1kb_p.bed</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">promoters <span class="keyword">for</span> non-protein-coding genes</span></span><br><span class="line">zcat gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; $3==&quot;transcript&quot; &#123;print $1,$4-1,$5,$18,&quot;promoter(NP)&quot;,$7,$14&#125;&#x27; | tr -d &#x27;&quot;;&#x27; | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;if ($7!=&quot;protein_coding&quot;)&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;102,194,165&quot;&#125;&#125;&#x27; | \</span><br><span class="line">    bedtools flank -i - -g hg38.genome -l 1000 -r 0 -s &gt; promoters_1kb_np.bed</span><br></pre></td></tr></table></figure></li><li><p>File name: <code>gencode.v24.segmented.tssflanking500b.bed.gz</code></p></li><li><p>Local Path: <em>&#x2F;References&#x2F;Anotations&#x2F;human&#x2F;hg38&#x2F;gencode.v24.segmented.tss*<em>flanking500b</em></em>.bed.gz*</p></li><li><p>Remote backup: <a target="_blank" rel="noopener external nofollow noreferrer" href="https://osf.io/4ht6c/">OSF</a></p></li><li><p>Description: Genomic segmentations based on GENCODE v24, promoters were defined as TSS $\pm$ 500bp (transcripts). (<code>2e624c3bc2330beb81464558ead1a11e</code>)</p></li><li><p>Recipe:</p><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_"># </span><span class="language-bash">promoters <span class="keyword">for</span> protein-coding genes</span></span><br><span class="line">zcat gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; $3==&quot;transcript&quot; &#123;print $1,$4-1,$5,$18,&quot;promoter&quot;,$7,$14&#125;&#x27; | tr -d &#x27;&quot;;&#x27; | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;if ($7==&quot;protein_coding&quot;)&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;102,194,165&quot;&#125;&#125;&#x27; | \</span><br><span class="line">    bedtools flank -i - -g hg38.genome -l 500 -r 0 -s | \</span><br><span class="line">    bedtools slop -i - -g hg38.genome -l 0 -r 500 -s &gt; promoters_500bp.bed</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">promoters <span class="keyword">for</span> non-protein-coding genes</span></span><br><span class="line">zcat gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; $3==&quot;transcript&quot; &#123;print $1,$4-1,$5,$18,&quot;promoter(NP)&quot;,$7,$14&#125;&#x27; | tr -d &#x27;&quot;;&#x27; | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;if ($7!=&quot;protein_coding&quot;)&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;102,194,165&quot;&#125;&#125;&#x27; | \</span><br><span class="line">    bedtools flank -i - -g hg38.genome -l 500 -r 0 -s \</span><br><span class="line">    bedtools slop -i - -g hg38.genome -l 0 -r 500 -s &gt; np_promoters_500bp.bed</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">intergenic</span></span><br><span class="line">zcat gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;&#125; $3==&quot;gene&quot; &#123;print $1,$4-1,$5,$10,$16,$7&#125;&#x27; | \</span><br><span class="line">    tr -d &#x27;&quot;;&#x27; | \</span><br><span class="line">    bedtools slop -i - -g hg38.genome -l 500 -r 0 -s | \</span><br><span class="line">    sortBed -g ../hg38.genome | \</span><br><span class="line">    bedtools complement -i stdin -g ../hg38.genome | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;print $1,$2,$3,&quot;.&quot;,&quot;intergenic&quot;,&quot;.&quot;,$2,$3,&quot;141,160,203&quot;&#125;&#x27; &gt; intergenic_500bp.bed</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">exons</span></span><br><span class="line">zcat gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;&#125; $3==&quot;exon&quot; &#123;print $1,$4-1,$5,$18,&quot;exon&quot;,$7&#125;&#x27; | \</span><br><span class="line">    tr -d &#x27;&quot;;&#x27; | \</span><br><span class="line">    sortBed -g ../hg38.genome | \</span><br><span class="line">    mergeBed -i - -c 4,5,6 -o distinct,distinct,distinct -s | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;231,138,195&quot;&#125;&#x27; &gt; exons.bed</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">introns</span></span><br><span class="line">zcat gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;&#125; $3==&quot;gene&quot; &#123;print $1,$4-1,$5,$16,&quot;intron&quot;,$7&#125;&#x27; | \</span><br><span class="line">    tr -d &#x27;&quot;;&#x27; | \</span><br><span class="line">    sortBed -g ../hg38.genome | \</span><br><span class="line">    subtractBed -a stdin -b exons.bed | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;255,217,47&quot;&#125;&#x27; &gt; introns.bed</span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">UTR, perl script from https://davetang.org/muse/2012/09/12/gencode/</span></span><br><span class="line">get_35_utr.pl gencode.v24.annotation.gtf.gz | \</span><br><span class="line">    awk &#x27;BEGIN&#123;OFS=&quot;\t&quot;;FS=&quot;\t&quot;&#125;&#123;if ($5==&quot;3_UTR&quot;)&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;166,216,84&quot;&#125;else&#123;print $1,$2,$3,$4,$5,$6,$2,$3,&quot;252,141,98&quot;&#125;&#125;&#x27; &gt; utr.bed</span><br><span class="line"></span><br><span class="line">cat intergenic_500bp.bed promoters_500bp.bed np_promoters_500bp.bed utr.bed introns.bed exons.bed | sort -k1,1 -k2,2n | bgzip &gt; gencode.v24.segmented.tssflanking500b.bed.gz</span><br></pre></td></tr></table></figure></li></ul><h5 id="RefGene"><a href="#RefGene" class="headerlink" title="RefGene"></a>RefGene</h5><ul><li>File name: <code>refseq.ver109.20190125.annotation.gtf.gz</code> (MD5 checksum: <code>848813de5b516e0f328046ef9c931091</code>)</li><li>Local path: <em>&#x2F;References&#x2F;Annotations&#x2F;human&#x2F;hg38&#x2F;refseq.ver109.20190125.annotation.gtf.gz</em></li><li>Remote backup: <a target="_blank" rel="noopener external nofollow noreferrer" href="https://osf.io/9jk3e/">OSF</a></li><li>Description: RefSeq annotation in GTF format that has been remapped to use the same set of UCSC-style sequence identifiers used in the FASTA files. The annotation is NCBI Homo sapiens Updated Annotation Release 109.20190125 from 25 January 2019.</li><li>Recipe:<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/GCA_000001405.15_GRCh38_full_analysis_set.refseq_annotation.gtf.gz</span><br><span class="line">mv GCA_000001405.15_GRCh38_full_analysis_set.refseq_annotation.gtf.gz refseq.ver109.20190125.annotation.gtf.gz</span><br></pre></td></tr></table></figure></li></ul><h4 id="Other-annotations"><a href="#Other-annotations" class="headerlink" title="Other annotations"></a>Other annotations</h4><h5 id="Repeat-Masker"><a href="#Repeat-Masker" class="headerlink" title="Repeat Masker"></a>Repeat Masker</h5><ul><li>File name: <code>rmsk.bed.gz</code> (MD5 checksum: <code>ae12aefbef9d4f5bc7695158a67d9a55</code>)</li><li>Local path: <em>&#x2F;References&#x2F;Annotations&#x2F;human&#x2F;hg38&#x2F;rmsk.bed.gz</em></li><li>Remote backup: <a target="_blank" rel="noopener external nofollow noreferrer" href="https://osf.io/e7w3k/">OSF</a></li><li>Description: Repeat Masker from UCSC. The following fields were selected:<ul><li>genoName (Genomic sequence name)</li><li>genoStart (Start in genomic sequence)</li><li>genoEnd (End in genomic sequence)</li><li>strand (Relative orientation + or -)</li><li>repName (Name of repeat)</li><li>repFamily (Family of repeat).</li></ul></li><li>Recipe:<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">wget http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/rmsk.txt.gz</span><br><span class="line">gunzip rmsk.txt.gz</span><br><span class="line">gawk &#x27;OFS=&quot;\t&quot;&#123;print $6,$7,$8,$11,$13,$10&#125;&#x27; rmsk.txt | \</span><br><span class="line">    sort -k1,1 -k2,2n | \</span><br><span class="line">    bgzip &gt; rmsk.bed.gz</span><br></pre></td></tr></table></figure></li></ul><h2 id="Generic"><a href="#Generic" class="headerlink" title="Generic"></a>Generic</h2><h3 id="Sequences-1"><a href="#Sequences-1" class="headerlink" title="Sequences"></a>Sequences</h3><ul><li>Primary assembly:</li><li>rRNA: Human ribosomal DNA complete repeating unit, GenBank accession code: U13369.1 .</li></ul><h3 id="Annotations-1"><a href="#Annotations-1" class="headerlink" title="Annotations"></a>Annotations</h3><h4 id="Motif-databases-MEME"><a href="#Motif-databases-MEME" class="headerlink" title="Motif databases (MEME)"></a>Motif databases (MEME)</h4><ul><li>File name: <code>motif_databases.12.19.tgz</code> (MD5 checksum: <code>f5ffcaecc07570ee19dba20b82d7bd73</code>)</li><li>Local path: <em>&#x2F;References&#x2F;Annotations&#x2F;human&#x2F;generic&#x2F;motif_databases.12.19.tgz</em></li><li>Remote backup: <a target="_blank" rel="noopener external nofollow noreferrer" href="https://osf.io/cktus/">OSF</a></li><li>Description: Motif databases for MEME suite (updated 28 Oct 2019).</li><li>Recipe:<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wget http://alternate.meme-suite.org/meme-software/Databases/motifs/motif_databases.12.19.tgz</span><br></pre></td></tr></table></figure></li></ul><h1 id="Note"><a href="#Note" class="headerlink" title="Note"></a>Note</h1><ul><li>For all fasta files, 3 standard annotations will also be generated simultaneously:<ul><li><code>.fai</code>: index which allows for fast and random access to any sequences in the indexed fasta file. This index is generated with the following command:<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools faidx input.fa</span><br></pre></td></tr></table></figure></li><li><code>.genome</code>: Table with two columns, specifying length of each chromosome.<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cut -f1,2 input.fa.fai &gt; size.genome</span><br></pre></td></tr></table></figure></li><li><code>.dict</code>:<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">java -jar picard.jar CreateSequenceDictionary \</span><br><span class="line">    R=input.fa \ </span><br><span class="line">    O=input.dict</span><br></pre></td></tr></table></figure></li></ul></li><li>There are two types of promoters in both <code>gencode.v24.segmented.tssup1kb.bed</code> and <code>gencode.v24.segmented.tssflanking1kb.bed</code> :<ul><li>Promoters for protein coding genes (denote as <code>promoter</code> in these files)</li><li>Promoters for non-protein coding genes (denote as <code>promoter(NP)</code>)</li></ul></li></ul></div><blockquote class="post-copyright"><div class="content"><div> <span class="post-time">Last updated: <time datetime="2025-04-23T04:03:22.945Z" itemprop="dateUpdated">2025-04-23 04:03:22</time></span><br></div><div>This article is posted in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.</div><div>Permanent link to this post: <a href="https://www.yaobio.com/posts/collection-of-commonly-used-references-in-bioinformatic-analysis/">https://www.yaobio.com/posts/collection-of-commonly-used-references-in-bioinformatic-analysis/</a></div></div><footer> <a href="https://www.yaobio.com"><img src="/img/profile.png" alt="Li Yao"> Li Yao</a></footer></blockquote><div class="post-footer"></div></div><nav class="post-nav flex-row flex-justify-between"><div class="waves-block waves-effect prev"><a href="/posts/file-templates-for-writing-codes/" id="post-prev" class="post-nav-link"><div class="tips"><i class="icon fa-solid fa-angle-left fa-lg icon-pr"></i> Prev</div><h4 class="title">File templates for writing codes</h4></a></div></nav><section class="comments" id="comments"><div id="disqus_thread"></div><script>var disqus_shortname="yaobio";lazyScripts.push("//"+disqus_shortname+".disqus.com/embed.js")</script><noscript>Please enable JavaScript to view the <a target="_blank" rel="noopener external nofollow noreferrer" href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript></section></article></div><footer class="footer"><div class="top"><p> <span>This page is available under the <a rel="license noopener" target="_blank" href="//creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</span></p></div><div class="bottom"><p><span>Li Yao &copy; 2019 - 2025</span> <span>Modified from indigo.</span></p></div></footer></main><div class="mask" id="mask"></div><a href="javascript:;" rel="external nofollow noreferrer" id="gotop" class="waves-effect waves-circle waves-light"><span class="fa-solid fa-lg fa-chevron-up"></span></a><script src="/js/waves.min.js?v=2.0.0"></script><script>MathJax={tex:{inlineMath:[["$","$"],["\\(","\\)"]]},svg:{fontCache:"global"}}</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><script>var BLOG={ROOT:"/",SHARE:!1,REWARD:!1}</script><script src="/js/main.js"></script><div class="search-panel" id="search-panel"><ul class="search-result" id="search-result"></ul></div><template id="search-tpl"><li class="item"><a href="{path}" class="waves-block waves-effect"><div class="title ellipsis" title="{title}">{title}</div><div class="flex-row flex-middle"><div class="tags ellipsis"> {tags}</div> <time class="flex-col time">{date}</time></div></a></li></template><script src="/js/search.min.js?v=2.0.0" async></script></body></html>