<!DOCTYPE html><html><head><meta charset="utf-8"><title>PINTS and PINTS web portal | Li Yao</title><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1"><meta name="theme-color" content="#3F51B5"><meta name="keywords" content=""><meta name="description" content="Peak Identifier for Nascent Transcript Starts (PINTS) is a software that helps  locate potential regulatory elements (promoters and enhancers) genome-wide from nascent transcript sequencing data, like"><meta property="og:type" content="article"><meta property="og:title" content="PINTS and PINTS web portal"><meta property="og:url" content="https://www.yaobio.com/tools/pints-and-pints-web-portal/index.html"><meta property="og:site_name" content="Li Yao"><meta property="og:description" content="Peak Identifier for Nascent Transcript Starts (PINTS) is a software that helps  locate potential regulatory elements (promoters and enhancers) genome-wide from nascent transcript sequencing data, like"><meta property="og:locale" content="en_US"><meta property="article:published_time" content="2021-03-07T23:44:11.000Z"><meta property="article:modified_time" content="2025-04-23T04:03:22.948Z"><meta property="article:author" content="Li Yao"><meta name="twitter:card" content="summary"><link rel="alternate" type="application/atom+xml" title="Li Yao" href="/atom.xml"><link rel="shortcut icon" href="/favicon.ico"><link rel="stylesheet" href="/css/style.css?v=2.0.0"><link rel="stylesheet" href="/css/li.css?v=2.0.0"><script>window.lazyScripts=[]</script><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><meta name="generator" content="Hexo 6.2.0"></head><body><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><div id="loading" class="active"></div><aside id="menu" class="hide"><div class="inner flex-row-vertical"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="menu-off"><i class="fa-solid fa-lg fa-xmark"></i></a><div class="brand-wrap" style="background-image:url(/img/brand.jpg)"><div class="brand"> <a href="/" class="avatar waves-effect waves-circle waves-light"><img src="/img/profile.png"></a><hgroup class="introduce"><h5 class="nickname"> Li Yao</h5></hgroup></div></div><div class="scroll-wrap flex-col"><ul class="nav"><li class="waves-block waves-effect"><a href="/"><i class="icon fa-solid fa-lg fa-house-chimney"></i> Home</a></li><li class="waves-block waves-effect"><a href="/publications"><i class="icon fa-solid fa-lg fa-book"></i> Publications</a></li><li class="waves-block waves-effect active"><a href="/tools"><i class="icon fa-solid fa-lg fa-rocket"></i> Tools</a></li><li class="waves-block waves-effect"><a href="/teaching"><i class="icon fa-solid fa-lg fa-person-chalkboard"></i> Teaching</a></li></ul></div></div></aside><main id="main"><div class="uni-header"><header class="top-header" id="header"><div class="flex-row"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light on" id="menu-toggle"><i class="fa-solid fa-lg fa-bars"></i></a><div class="flex-col header-title ellipsis">PINTS and PINTS web portal</div><div class="search-wrap" id="search-wrap"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="back"><i class="fa-solid fa-lg fa-chevron-left"></i></a> <input type="text" id="key" class="search-input" autocomplete="off" placeholder="Search"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="search"><i class="fa-solid fa-lg fa-magnifying-glass"></i></a></div></div></header><header class="content-header post-header"><div class="container fade-scale"><h1 class="title">PINTS and PINTS web portal</h1><h5 class="subtitle"> <time datetime="2021-03-07T23:44:11.000Z" itemprop="datePublished" class="page-time">2021-03-07</time><ul class="article-category-list"><li class="article-category-list-item"><a class="article-category-list-link" href="/tools/">tools</a></li></ul></h5></div></header> <img aria-hidden="true" class="bkg-img" src="/assets/showcases/pints.png"></div><div class="container body-wrap"><aside class="post-widget"><nav class="post-toc-wrap post-toc-shrink" id="post-toc"><h4>Table of Content</h4><ol class="post-toc"><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Summary"><span class="post-toc-number">1.</span> <span class="post-toc-text">Summary</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Availability"><span class="post-toc-number">2.</span> <span class="post-toc-text">Availability</span></a></li><li class="post-toc-item post-toc-level-2"><a class="post-toc-link" href="#Changelog"><span class="post-toc-number">3.</span> <span class="post-toc-text">Changelog</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#PINTS"><span class="post-toc-number">3.1.</span> <span class="post-toc-text">PINTS</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-15"><span class="post-toc-number">3.1.1.</span> <span class="post-toc-text">1.1.15</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-14"><span class="post-toc-number">3.1.2.</span> <span class="post-toc-text">1.1.14</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-13"><span class="post-toc-number">3.1.3.</span> <span class="post-toc-text">1.1.13</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-12"><span class="post-toc-number">3.1.4.</span> <span class="post-toc-text">1.1.12</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-11"><span class="post-toc-number">3.1.5.</span> <span class="post-toc-text">1.1.11</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-10"><span class="post-toc-number">3.1.6.</span> <span class="post-toc-text">1.1.10</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-9"><span class="post-toc-number">3.1.7.</span> <span class="post-toc-text">1.1.9</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-8"><span class="post-toc-number">3.1.8.</span> <span class="post-toc-text">1.1.8</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-7"><span class="post-toc-number">3.1.9.</span> <span class="post-toc-text">1.1.7</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-6"><span class="post-toc-number">3.1.10.</span> <span class="post-toc-text">1.1.6</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-5"><span class="post-toc-number">3.1.11.</span> <span class="post-toc-text">1.1.5</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-4"><span class="post-toc-number">3.1.12.</span> <span class="post-toc-text">1.1.4</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#1-1-0-1-1-3"><span class="post-toc-number">3.1.13.</span> <span class="post-toc-text">1.1.0 (1.1.3)</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#0-0-1-post0-dev8"><span class="post-toc-number">3.1.14.</span> <span class="post-toc-text">0.0.1.post0.dev8</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#0-0-1-post0-dev7"><span class="post-toc-number">3.1.15.</span> <span class="post-toc-text">0.0.1.post0.dev7</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#0-0-1-post0-dev6"><span class="post-toc-number">3.1.16.</span> <span class="post-toc-text">0.0.1.post0.dev6</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#0-0-1-post0-dev5"><span class="post-toc-number">3.1.17.</span> <span class="post-toc-text">0.0.1.post0.dev5</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#0-0-1-post0-dev3"><span class="post-toc-number">3.1.18.</span> <span class="post-toc-text">0.0.1.post0.dev3</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#0-0-1-post0-dev2"><span class="post-toc-number">3.1.19.</span> <span class="post-toc-text">0.0.1.post0.dev2</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#0-0-1-post0-dev1"><span class="post-toc-number">3.1.20.</span> <span class="post-toc-text">0.0.1.post0.dev1</span></a></li></ol></li><li class="post-toc-item post-toc-level-3"><a class="post-toc-link" href="#PINTS-web-portal"><span class="post-toc-number">3.2.</span> <span class="post-toc-text">PINTS web portal</span></a><ol class="post-toc-child"><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#2023v1"><span class="post-toc-number">3.2.1.</span> <span class="post-toc-text">2023v1</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#2022v1"><span class="post-toc-number">3.2.2.</span> <span class="post-toc-text">2022v1</span></a></li><li class="post-toc-item post-toc-level-4"><a class="post-toc-link" href="#2021v1"><span class="post-toc-number">3.2.3.</span> <span class="post-toc-text">2021v1</span></a></li></ol></li></ol></li></ol></nav></aside><article id="post-tools/PINTS" class="post-article article-type-post fade" itemprop="blogPost"><div class="post-card"><h1 class="post-card-title">PINTS and PINTS web portal</h1><div class="post-meta"> <time class="post-time" title="2021-03-07 23:44:11" datetime="2021-03-07T23:44:11.000Z" itemprop="datePublished">2021-03-07</time><span id="busuanzi_container_page_pv" title="Views" style="display:none"><i class="icon fa-solid fa-eye icon-pr"></i><span id="busuanzi_value_page_pv"></span></span></div><div class="post-content" id="post-content" itemprop="postContent"><blockquote><p>If you are interested in the detailed methodology, please refer to our manuscript, <em><a href="/publication/a-comparison-of-experimental-assays-and-analytical-methods-for-genome-wide-identification-of-active-enhancers/" title="A comparison of experimental assays and analytical methods for genome-wide identification of active enhancers">A comparison of experimental assays and analytical methods for genome-wide identification of active enhancers</a></em>, for more information.</p></blockquote><h2 id="Summary"><a href="#Summary" class="headerlink" title="Summary"></a>Summary</h2><p><strong>P</strong>eak <strong>I</strong>dentifier for <strong>N</strong>ascent <strong>T</strong>ranscript <strong>S</strong>tarts (PINTS) is a software that helps locate potential regulatory elements (promoters and enhancers) genome-wide from nascent transcript sequencing data, like GRO&#x2F;PRO-cap libraries, which are specifically enriched for signals from <strong>t</strong>ranscription <strong>s</strong>tart <strong>s</strong>ites (TSSs).</p><p><object data="pints_intro.svg" width="50%" class="svg-animation"></object></p><p>PINTS web portal is a comprehensive catalog of promoters and enhancers, and all have support from transcriptional evidence. Besides the catalog database, the portal also allows users to upload their genomic loci of interest and analyze the loci with state-of-the-art annotations. The portal is a containerized web service powered by Django, celery, and other open-source packages.</p><h2 id="Availability"><a href="#Availability" class="headerlink" title="Availability"></a>Availability</h2><p>PINTS executable files are packed and distributed on both <a target="_blank" rel="noopener external nofollow noreferrer" href="https://pypi.org/project/pyPINTS/">PyPI</a> and <a target="_blank" rel="noopener external nofollow noreferrer" href="https://bioconda.github.io/recipes/pypints/README.html#package-pypints">bioconda</a>. You can install PINTS like any other standard Python package with <a target="_blank" rel="noopener external nofollow noreferrer" href="https://pip.pypa.io/en/stable/">pip</a> (<code>pip install pyPINTS</code>) or <a target="_blank" rel="noopener external nofollow noreferrer" href="https://docs.conda.io/en/latest/">conda</a> (<code>conda install -c bioconda pypints</code>). The source code is available on <a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/hyulab/pints">GitHub</a>.</p><p>PINTS web portal is freely accessible at <a target="_blank" rel="noopener external nofollow noreferrer" href="https://pints.yulab.org/">pints.yulab.org</a>.</p><h2 id="Changelog"><a href="#Changelog" class="headerlink" title="Changelog"></a>Changelog</h2><h3 id="PINTS"><a href="#PINTS" class="headerlink" title="PINTS"></a>PINTS</h3><h4 id="1-1-15"><a href="#1-1-15" class="headerlink" title="1.1.15"></a>1.1.15</h4><p>Release date: 2024-12-22</p><ul><li><strong>[Fixed]</strong>: PINTS (boundary extender) may quit when warnings are threw in third-party libraries.</li></ul><h4 id="1-1-14"><a href="#1-1-14" class="headerlink" title="1.1.14"></a>1.1.14</h4><p>Release date: 2024-10-10</p><ul><li><strong>[Fixed]</strong>: some csi index files not deleted after the entire peak calling process (related to ver<a href="#1-1-12">1.1.12</a>).</li></ul><h4 id="1-1-13"><a href="#1-1-13" class="headerlink" title="1.1.13"></a>1.1.13</h4><p>Release date: 2024-09-09</p><ul><li><strong>[Fixed]</strong>: PINTS may quit in cases where the sets of chromosomes in the forward and reverse bigwigs are not 100% identical</li></ul><h4 id="1-1-12"><a href="#1-1-12" class="headerlink" title="1.1.12"></a>1.1.12</h4><p>Release date: 2024-07-21</p><ul><li><strong>[Changed]</strong>: improved IO performance by up to ~40% when working with bigWig inputs</li><li><strong>[Changed]</strong>: years in comments</li><li><strong>[Added]</strong>: pints_counter for generating count matrix</li><li><strong>[Fixing]</strong>: switched to csi index for better handling of large chromosomes. But more tests regarding memory pressure are needed</li><li><strong>[Fixed]</strong>: csi index files not deleted after the entire peak calling process</li></ul><h4 id="1-1-11"><a href="#1-1-11" class="headerlink" title="1.1.11"></a>1.1.11</h4><p>Skipped</p><h4 id="1-1-10"><a href="#1-1-10" class="headerlink" title="1.1.10"></a>1.1.10</h4><p>Release date: 2023-11-07</p><ul><li><strong>[Changed]</strong>: Removed warning filters in extension_engine as warnings from other packages may break downstream analysis.</li><li><strong>[Changed]</strong>: years in comments</li></ul><h4 id="1-1-9"><a href="#1-1-9" class="headerlink" title="1.1.9"></a>1.1.9</h4><p>Release date: 2023-05-11</p><ul><li><strong>[Added]</strong>: A new QC script for evaluating TSS enrichment (over gene body regions)</li><li><strong>[Changed]</strong>: Report scale factors when using <code>pints_visualizer</code></li><li><strong>[Changed]</strong>: Switched <code>formatter_class</code> for all executable scripts’ argument parsers from <code>HelpFormatter</code> to <code>ArgumentDefaultsHelpFormatter</code>, so that default values will be printed when running commands like <code>pints_caller --help</code></li><li><strong>[Fixed]</strong>: Added boundary caps to avoid <code>ValueError</code> when calling <code>pysam.TabixFile.fetch</code> (Corresponding to GitHub issue #6)</li></ul><h4 id="1-1-8"><a href="#1-1-8" class="headerlink" title="1.1.8"></a>1.1.8</h4><p>Release date: 2022-10-08</p><ul><li><strong>[Fixed]</strong>: When no divergent&#x2F;bidirectional peaks are detected, PINTS may throw exceptions. In this version, empty predictions will be skipped, so errors like <code>ERROR: Requested column 12, but database file stdin only has fields 1 - 0.</code> will be avoided (related GitHub issue <a target="_blank" rel="noopener external nofollow noreferrer" href="https://github.com/hyulab/PINTS/issues/12">#12</a>).</li><li><strong>[Changed]</strong>: Since PINTS will automatically adjust the value of <code>--min-mu-percent</code>, now it prints info instead of a warning message when <code>--min-mu-percent</code> related issues are detected.</li><li><strong>[Changed]</strong>: Reduce redundancy in the function <code>peak_calling</code> by abstracting statements about parsing input files.</li><li><strong>[Added]</strong>: Added a new QC metric on the number of significant calls. If the number of significant calls is too large, PINTS will throw a warning message and suggest users switch to a more strict FDR cutoff.</li></ul><h4 id="1-1-7"><a href="#1-1-7" class="headerlink" title="1.1.7"></a>1.1.7</h4><p>Release date: 2022-09-16</p><ul><li><strong>[Added]</strong>: unit test cases</li><li><strong>[Added]</strong>: Github issue templates</li><li><strong>[Changed]</strong>: Improved the logic of on-the-fly QC. If QC is enabled, now PINTS tries to directly adjust corresponding parameter (for now only <code>--min-mu-percent</code>) and log the changes. This behavior can be repressed by using <code>--disable-qc</code>.</li></ul><h4 id="1-1-6"><a href="#1-1-6" class="headerlink" title="1.1.6"></a>1.1.6</h4><p>Release date: 2022-07-05</p><ul><li><strong>[Added]</strong>: Added chromosome length check when using BAM files as inputs</li><li><strong>[Changed]</strong>: Simplified on-the-fly QC messages. So warning messages of the same type only show up once at the end of the log file.</li><li><strong>[Changed]</strong>: Dropped peak length limits for empirical LER.</li><li><strong>[Fixed]</strong>: Fixed inappropriate usage of <code>str.find</code> in <code>annotate_tre_with_epig_info</code>.</li></ul><h4 id="1-1-5"><a href="#1-1-5" class="headerlink" title="1.1.5"></a>1.1.5</h4><p>Release date: 2022-06-16</p><ul><li><strong>[Fixed]</strong>: Add explicit type conversion to avoid <code>AttributeError: Can only use str accessor with string values</code></li></ul><h4 id="1-1-4"><a href="#1-1-4" class="headerlink" title="1.1.4"></a>1.1.4</h4><p>Release date: 2022-06-13</p><ul><li><strong>[Fixed]</strong>: fixed typos in readme;</li><li><strong>[Removed]</strong>: Removed scaffold codes;</li><li><strong>[Changed]</strong>: changed default output from pints_boundary_extender to standard bed format</li><li><strong>[Changed]</strong>: moved codes from scripts to PINTS lib for more accurate test coverage analysis;</li></ul><h4 id="1-1-0-1-1-3"><a href="#1-1-0-1-1-3" class="headerlink" title="1.1.0 (1.1.3)"></a>1.1.0 (1.1.3)</h4><p>Release date: 2022-03-16 (for 1.1.0) and 2022-03-21 (for 1.1.3)</p><ul><li><strong>[Added]</strong>: Source distribution to CI&#x2F;CD for creating conda packages</li><li><strong>[Added]</strong>: Added <code>MANIFEST.in</code> file to include <code>versioneer.py</code> in the source distribution</li><li><strong>[Added]</strong>: empirical LER</li><li><strong>[Added]</strong>: On the fly QC</li><li><strong>[Changed]</strong>: Switched versioning schema from PEP 440 to <code>major_ver</code>.<code>middle_ver</code>.<code>minor_ver</code>. The first two vers will be defined by tags; the <code>minor_ver</code> will be inferred automatically from commit distance.</li><li><strong>[Changed]</strong>: Changed the default value for <code>--alpha</code> (for combining nearby peaks) to 0.3.</li><li><strong>[Changed]</strong>: Renamed <code>README.md</code>.</li><li><strong>[Changed]</strong>: Improved the implementation of independent filtering</li><li><strong>[Changed]</strong>: Improved code readability with Pylint</li><li><strong>[Removed]</strong>: Scaffolding parameters and simplified naming schema.</li></ul><h4 id="0-0-1-post0-dev8"><a href="#0-0-1-post0-dev8" class="headerlink" title="0.0.1.post0.dev8"></a>0.0.1.post0.dev8</h4><p>Release date: 2022-01-07</p><ul><li><strong>[Changed]</strong>: Add requests to <code>install_requires</code>.</li></ul><h4 id="0-0-1-post0-dev7"><a href="#0-0-1-post0-dev7" class="headerlink" title="0.0.1.post0.dev7"></a>0.0.1.post0.dev7</h4><p>Release date: 2022-01-07</p><ul><li><strong>[Changed]</strong>: Updated readme and help message</li></ul><h4 id="0-0-1-post0-dev6"><a href="#0-0-1-post0-dev6" class="headerlink" title="0.0.1.post0.dev6"></a>0.0.1.post0.dev6</h4><p>Release date: 2022-01-05</p><ul><li><strong>[Fixed]</strong>: Fixed timestamp issue when creating log file</li></ul><h4 id="0-0-1-post0-dev5"><a href="#0-0-1-post0-dev5" class="headerlink" title="0.0.1.post0.dev5"></a>0.0.1.post0.dev5</h4><p>Release date: 2022-01-04</p><ul><li><strong>[Added]</strong>: Introduced version check, which prints update hints if current version of PINTS is out-dated.</li></ul><h4 id="0-0-1-post0-dev3"><a href="#0-0-1-post0-dev3" class="headerlink" title="0.0.1.post0.dev3"></a>0.0.1.post0.dev3</h4><p>Release date: 2021-10-30</p><ul><li><strong>[Changed]</strong>: Added pyBigWig, biopython, and matplotlib into <code>setup.install_requires</code>.</li><li><strong>[Added]</strong>: Introduced a new module for refining peak calls with epigenomic evidence from PINTS webserver. This feature can be activated by using <code>--epig-annotation &lt;biosample_name&gt;</code>.</li><li><strong>[Added]</strong>: error check for lowly sequenced libraries.</li></ul><h4 id="0-0-1-post0-dev2"><a href="#0-0-1-post0-dev2" class="headerlink" title="0.0.1.post0.dev2"></a>0.0.1.post0.dev2</h4><p>Release date: 2021-08-04</p><ul><li><strong>[Added]</strong>: Configured CI&#x2F;CD for uploading PINTS to PyPI.</li><li><strong>[Added]</strong>: Added CI&#x2F;CD badge in README.</li></ul><h4 id="0-0-1-post0-dev1"><a href="#0-0-1-post0-dev1" class="headerlink" title="0.0.1.post0.dev1"></a>0.0.1.post0.dev1</h4><p>Release date: 2021-06-03</p><p>Initial implementation.</p><h3 id="PINTS-web-portal"><a href="#PINTS-web-portal" class="headerlink" title="PINTS web portal"></a>PINTS web portal</h3><h4 id="2023v1"><a href="#2023v1" class="headerlink" title="2023v1"></a>2023v1</h4><ul><li><strong>[Added]</strong>: Included PRO-cap libraries for HCT116</li><li><strong>[Changed]</strong>: Updated epigenomic annotation to cCRE v4</li><li><strong>[Changed]</strong>: Updated functional characterization info from NCBI RefSeq</li><li><strong>[Changed]</strong>: Updated TF binding site annotation to JASPAR 2024</li></ul><h4 id="2022v1"><a href="#2022v1" class="headerlink" title="2022v1"></a>2022v1</h4><p>Release date: 2022-07-17</p><ul><li><strong>[Added]</strong>: Increased biosample coverage by adding PRO-cap libraries for:<ul><li>GM18507, GM19238 from Kristjánsdóttir <em>et al.</em> 2020 (DOI: 10.1038&#x2F;s41467-020-19829-z)</li><li>MCF7 from Hou <em>et al.</em> 2022 (DOI: 10.1016&#x2F;j.celrep.2022.110944)</li><li>A673, Caco-2, and MCF 10A from the ENCODE portal ( using DOI: 10.1038&#x2F;s41587-022-01211-7 as a placeholder for now, will update the study link after the manuscript is published)</li></ul></li><li><strong>[Changed]</strong>: Updated epigenomic annotation to cCRE v3</li><li><strong>[Changed]</strong>: Updated TF binding site annotation to JASPAR 2022</li><li><strong>[Changed]</strong>: Updated functional characterization info for HepG2 by including consistent enhancer calls from Klein <em>et al.</em> 2020.</li><li><strong>[Changed]</strong>: Switched to PINTS <a href="#1-1-6">1.1.6</a>.</li><li><strong>[Deprecated]</strong>: Removed bidirectional elements identified from RAMPAGE&#x2F;CAGE libraries that overlap with protein-coding exons (except $5^\prime$ UTR).</li></ul><h4 id="2021v1"><a href="#2021v1" class="headerlink" title="2021v1"></a>2021v1</h4><p>Release date: 2021-10-23</p><p>Initial release.</p></div><blockquote class="post-copyright"><div class="content"><div> <span class="post-time">Last updated: <time datetime="2025-04-23T04:03:22.948Z" itemprop="dateUpdated">2025-04-23 04:03:22</time></span><br></div><div>This article is posted in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.</div><div>Permanent link to this post: <a href="https://www.yaobio.com/tools/pints-and-pints-web-portal/">https://www.yaobio.com/tools/pints-and-pints-web-portal/</a></div></div><footer> <a href="https://www.yaobio.com"><img src="/img/profile.png" alt="Li Yao"> Li Yao</a></footer></blockquote><div class="post-footer"></div></div><nav class="post-nav flex-row flex-justify-between flex-row-reverse"><div class="waves-block waves-effect next"><a href="/tools/biosyntax/" id="post-next" class="post-nav-link"><div class="tips">Next<i class="icon fa-solid fa-angle-right fa-lg icon-pl"></i></div><h4 class="title">bioSyntax</h4></a></div></nav><section class="comments" id="comments"><div id="disqus_thread"></div><script>var disqus_shortname="yaobio";lazyScripts.push("//"+disqus_shortname+".disqus.com/embed.js")</script><noscript>Please enable JavaScript to view the <a target="_blank" rel="noopener external nofollow noreferrer" href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript></section></article></div><footer class="footer"><div class="top"><p> <span>This page is available under the <a rel="license noopener" target="_blank" href="//creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</span></p></div><div class="bottom"><p><span>Li Yao &copy; 2019 - 2025</span> <span>Modified from indigo.</span></p></div></footer></main><div class="mask" id="mask"></div><a href="javascript:;" rel="external nofollow noreferrer" id="gotop" class="waves-effect waves-circle waves-light"><span class="fa-solid fa-lg fa-chevron-up"></span></a><script src="/js/waves.min.js?v=2.0.0"></script><script>MathJax={tex:{inlineMath:[["$","$"],["\\(","\\)"]]},svg:{fontCache:"global"}}</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><script>var BLOG={ROOT:"/",SHARE:!1,REWARD:!1}</script><script src="/js/main.js"></script><div class="search-panel" id="search-panel"><ul class="search-result" id="search-result"></ul></div><template id="search-tpl"><li class="item"><a href="{path}" class="waves-block waves-effect"><div class="title ellipsis" title="{title}">{title}</div><div class="flex-row flex-middle"><div class="tags ellipsis"> {tags}</div> <time class="flex-col time">{date}</time></div></a></li></template><script src="/js/search.min.js?v=2.0.0" async></script></body></html>