<!DOCTYPE html><html><head><meta charset="utf-8"><title>Publication | Li Yao</title><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1"><meta name="theme-color" content="#3F51B5"><meta name="keywords" content=""><meta property="og:type" content="website"><meta property="og:title" content="Li Yao"><meta property="og:url" content="https://www.yaobio.com/publication/index.html"><meta property="og:site_name" content="Li Yao"><meta property="og:locale" content="en_US"><meta property="article:author" content="Li Yao"><meta name="twitter:card" content="summary"><link rel="alternate" type="application/atom+xml" title="Li Yao" href="/atom.xml"><link rel="shortcut icon" href="/favicon.ico"><link rel="stylesheet" href="/css/style.css?v=2.0.0"><link rel="stylesheet" href="/css/li.css?v=2.0.0"><script>window.lazyScripts=[]</script><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><meta name="generator" content="Hexo 6.2.0"></head><body><script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="https://www.googletagmanager.com/gtm.js?id=GTM-KGJ9HPC",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script><div id="loading" class="active"></div><aside id="menu"><div class="inner flex-row-vertical"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="menu-off"><i class="fa-solid fa-lg fa-xmark"></i></a><div class="brand-wrap" style="background-image:url(/img/brand.jpg)"><div class="brand"> <a href="/" class="avatar waves-effect waves-circle waves-light"><img src="/img/profile.png"></a><hgroup class="introduce"><h5 class="nickname"> Li Yao</h5></hgroup></div></div><div class="scroll-wrap flex-col"><ul class="nav"><li class="waves-block waves-effect"><a href="/"><i class="icon fa-solid fa-lg fa-house-chimney"></i> Home</a></li><li class="waves-block waves-effect"><a href="/publications"><i class="icon fa-solid fa-lg fa-book"></i> Publications</a></li><li class="waves-block waves-effect"><a href="/tools"><i class="icon fa-solid fa-lg fa-rocket"></i> Tools</a></li><li class="waves-block waves-effect"><a href="/teaching"><i class="icon fa-solid fa-lg fa-person-chalkboard"></i> Teaching</a></li></ul></div></div></aside><main id="main"><div class="uni-header"><header class="top-header" id="header"><div class="flex-row"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light on" id="menu-toggle"><i class="fa-solid fa-lg fa-bars"></i></a><div class="flex-col header-title ellipsis">Publication</div><div class="search-wrap" id="search-wrap"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="back"><i class="fa-solid fa-lg fa-chevron-left"></i></a> <input type="text" id="key" class="search-input" autocomplete="off" placeholder="Search"><a href="javascript:;" rel="external nofollow noreferrer" class="header-icon waves-effect waves-circle waves-light" id="search"><i class="fa-solid fa-lg fa-magnifying-glass"></i></a></div></div></header><header class="content-header index-header"><div class="container fade-scale"><h1 class="title">Publication</h1><h5 class="subtitle"></h5></div></header></div><div class="container body-wrap"><ul class="post-list"><li class="post-list-item fade"><article id="post-publications/2025-deepdetails" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2025-04-06 14:42:06" datetime="2025-04-06T14:42:06.000Z" itemprop="datePublished">2025-04-06</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/high-resolution-reconstruction-of-cell-type-specific-transcriptional-regulatory-processes-from-bulk-sequencing-samples/">High-resolution reconstruction of cell-type specific transcriptional regulatory processes from bulk sequencing samples</a></h3><div class="post-content" id="post-content" itemprop="postContent"> <a href="/publication/high-resolution-reconstruction-of-cell-type-specific-transcriptional-regulatory-processes-from-bulk-sequencing-samples/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2025-STARRseq-benchmark" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2025-03-27 18:41:29" datetime="2025-03-27T18:41:29.000Z" itemprop="datePublished">2025-03-27</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/comprehensive-evaluation-of-diverse-massively-parallel-reporter-assays-to-functionally-characterize-human-enhancers-genome-wide/">Comprehensive Evaluation of Diverse Massively Parallel Reporter Assays to Functionally Characterize Human Enhancers Genome-wide</a></h3><div class="post-content" id="post-content" itemprop="postContent"> <a href="/publication/comprehensive-evaluation-of-diverse-massively-parallel-reporter-assays-to-functionally-characterize-human-enhancers-genome-wide/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2025-QUASARR-seq" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2025-03-15 18:41:29" datetime="2025-03-15T18:41:29.000Z" itemprop="datePublished">2025-03-15</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/simultaneous-measurement-of-intrinsic-promoter-and-enhancer-potential-reveals-principles-of-functional-duality-and-regulatory-reciprocity/">Simultaneous measurement of intrinsic promoter and enhancer potential reveals principles of functional duality and regulatory reciprocity</a></h3><div class="post-content" id="post-content" itemprop="postContent"> <a href="/publication/simultaneous-measurement-of-intrinsic-promoter-and-enhancer-potential-reveals-principles-of-functional-duality-and-regulatory-reciprocity/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2024-kmt2cd" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2025-01-13 18:41:29" datetime="2025-01-13T18:41:29.000Z" itemprop="datePublished">2025-01-13</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/loss-of-kmt2c-or-kmt2d-primes-urothelium-for-tumorigenesis-and-redistributes-kmt2a-menin-to-bivalent-promoters/">Loss of Kmt2c or Kmt2d primes urothelium for tumorigenesis and redistributes KMT2A-menin to bivalent promoters</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Members of the KMT2C&#x2F;D-KDM6A complex are recurrently mutated in urothelial carcinoma and in histologically normal urothelium. Here, using genetically engineered mouse models, we demonstrate that Kmt2c&#x2F;d knockout in the urothelium led to impaired differentiation, augmented responses to growth and inflammatory stimuli and sensitization to oncogenic transformation by carcinogen and oncogenes. Mechanistically, KMT2D localized to active enhancers and CpG-poor promoters that preferentially regulate the urothelial lineage program and Kmt2c&#x2F;d knockout led to diminished H3K4me1, H3K27ac and nascent RNA transcription at these sites, which leads to impaired differentiation. Kmt2c&#x2F;d knockout further led to KMT2A-menin redistribution from KMT2D localized enhancers to CpG-high and bivalent promoters, resulting in derepression of signal-induced immediate early genes. Therapeutically, Kmt2c&#x2F;d knockout upregulated epidermal growth factor receptor signaling and conferred vulnerability to epidermal growth factor receptor inhibitors. Together, our data posit that functional loss of Kmt2c&#x2F;d licenses a molecular ‘field effect’ priming histologically normal urothelium for oncogenic transformation and presents therapeutic vulnerabilities. <a href="/publication/loss-of-kmt2c-or-kmt2d-primes-urothelium-for-tumorigenesis-and-redistributes-kmt2a-menin-to-bivalent-promoters/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2023-ncv-review" class="article-card article-type-post" itemprop="blogPost" style="padding-bottom:16px"><div class="post-flex"><div class="post-thumbnail"> <img src="/assets/covers/arg_2023.png"></div><div class="post-wrap"><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/finding-needles-in-the-haystack-strategies-for-uncovering-noncoding-regulatory-variants/">Finding Needles in the Haystack: Strategies for Uncovering Noncoding Regulatory Variants</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Despite accumulating evidence implicating noncoding variants in human diseases, unraveling their functionality remains a significant challenge. Systematic annotations of the regulatory landscape and the growth of sequence variant data sets have fueled the development of tools and methods to identify causal noncoding variants and evaluate their regulatory effects. Here, we review the latest advances in the field and discuss potential future research avenues to gain a more in-depth understanding of noncoding regulatory variants. <a href="/publication/finding-needles-in-the-haystack-strategies-for-uncovering-noncoding-regulatory-variants/" class="post-more waves-effect waves-button">Continue reading...</a></div></div></div></article></li><li class="post-list-item fade"><article id="post-publications/2022-epi-review" class="article-card article-type-post" itemprop="blogPost" style="padding-bottom:16px"><div class="post-flex"><div class="post-thumbnail"> <img src="/assets/covers/hmg_2022.jpeg"></div><div class="post-wrap"><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/functional-genomic-assays-to-annotate-enhancer-promoter-interactions-genome-wide/">Functional genomic assays to annotate enhancer-promoter interactions genome-wide</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Enhancers are pivotal for regulating gene transcription that occurs at promoters. Identification of the interacting enhancer–promoter pairs and understanding the mechanisms behind how they interact and how enhancers modulate transcription can provide fundamental insight into gene regulatory networks. Recently, advances in high-throughput methods in three major areas—chromosome conformation capture assay, such as Hi-C to study basic chromatin architecture, ectopic reporter experiments such as self-transcribing active regulatory region sequencing (STARR-seq) to quantify promoter and enhancer activity, and endogenous perturbations such as clustered regularly interspaced short palindromic repeat interference (CRISPRi) to identify enhancer–promoter compatibility—have further our knowledge about transcription. In this review, we will discuss the major method developments and key findings from these assays. <a href="/publication/functional-genomic-assays-to-annotate-enhancer-promoter-interactions-genome-wide/" class="post-more waves-effect waves-button">Continue reading...</a></div></div></div></article></li><li class="post-list-item fade"><article id="post-publications/2022-rbp-preference" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2022-08-16 14:47:18" datetime="2022-08-16T14:47:18.000Z" itemprop="datePublished">2022-08-16</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/survey-of-the-binding-preferences-of-rna-binding-proteins-to-rna-editing-events/">Survey of the binding preferences of RNA-binding proteins to RNA editing events</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Background: Adenosine-to-inosine (A-to-I) editing is an important RNA posttranscriptional process related to a multitude of cellular and molecular activities. However, systematic characterizations of whether and how the events of RNA editing are associated with the binding preferences of RNA sequences to RNA-binding proteins (RBPs) are still lacking. Results: With the RNA-seq and RBP eCLIP-seq datasets from the ENCODE project, we quantitatively survey the binding preferences of 150 RBPs to RNA editing events, followed by experimental validations. Such analyses of the RBP-associated RNA editing at nucleotide resolution and genome-wide scale shed light on the involvement of RBPs specifically in RNA editing-related processes, such as RNA splicing, RNA secondary structures, RNA decay, and other posttranscriptional processes. Conclusions: These results highlight the relevance of RNA editing in the functions of many RBPs and therefore serve as a resource for further characterization of the functional associations between various RNA editing events and RBPs. <a href="/publication/survey-of-the-binding-preferences-of-rna-binding-proteins-to-rna-editing-events/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2022-prostate-procap" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2022-04-08 18:41:29" datetime="2022-04-08T18:41:29.000Z" itemprop="datePublished">2022-04-08</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/capped-nascent-rna-sequencing-reveals-novel-therapy-responsive-enhancers-in-prostate-cancer/">Capped nascent RNA sequencing reveals novel therapy-responsive enhancers in prostate cancer</a></h3><div class="post-content" id="post-content" itemprop="postContent"> <a href="/publication/capped-nascent-rna-sequencing-reveals-novel-therapy-responsive-enhancers-in-prostate-cancer/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li><li class="post-list-item fade"><article id="post-publications/2022-pints" class="article-card article-type-post" itemprop="blogPost" style="padding-bottom:16px"><div class="post-flex"><div class="post-thumbnail"> <img src="/assets/covers/nbt_2022_v40_i7.png"></div><div class="post-wrap"><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/a-comparison-of-experimental-assays-and-analytical-methods-for-genome-wide-identification-of-active-enhancers/">A comparison of experimental assays and analytical methods for genome-wide identification of active enhancers</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Mounting evidence supports the idea that transcriptional patterns serve as more specific identifiers of active enhancers than histone marks; however, the optimal strategy to identify active enhancers both experimentally and computationally has not been determined. Here, we compared 13 genome-wide RNA sequencing (RNA-seq) assays in K562 cells and show that nuclear run-on followed by cap-selection assay (GRO&#x2F;PRO-cap) has advantages in enhancer RNA detection and active enhancer identification. We also introduce a tool, peak identifier for nascent transcript starts (PINTS), to identify active promoters and enhancers genome wide and pinpoint the precise location of 5′ transcription start sites. Finally, we compiled a comprehensive enhancer candidate compendium based on the detected enhancer RNA (eRNA) transcription start sites (TSSs) available in 120 cell and tissue types, which can be accessed at https://pints.yulab.org. With knowledge of the best available assays and pipelines, this large-scale annotation of candidate enhancers will pave the way for selection and characterization of their functions in a timeand labor-efficient manner. <a href="/publication/a-comparison-of-experimental-assays-and-analytical-methods-for-genome-wide-identification-of-active-enhancers/" class="post-more waves-effect waves-button">Continue reading...</a></div></div></div></article></li><li class="post-list-item fade"><article id="post-publications/2019-dclk1" class="article-card article-type-post" itemprop="blogPost"><div class="post-meta"> <time class="post-time" title="2019-12-05 14:40:14" datetime="2019-12-05T14:40:14.000Z" itemprop="datePublished">2019-12-05</time></div><h3 class="post-title" itemprop="name"> <a class="post-title-link" href="/publication/inhibition-of-dclk1-down-regulates-pd-l1-expression-through-hippo-pathway-in-human-pancreatic-cancer/">Inhibition of DCLK1 down-regulates PD-L1 expression through Hippo pathway in human pancreatic cancer</a></h3><div class="post-content" id="post-content" itemprop="postContent"> Immunotherapy is one of the most promising strategies for cancer, compared with traditional treatments. As one of the key emerging immunotherapies, anti-PD-1&#x2F;PD-L1 treatment has brought survival benefits to many advanced cancer patients. However, in pancreatic cancer, immunotherapy-based approaches have not achieved a favorable clinical effect because of mismatch repair deficiencies. Therefore, the majority of pancreatic tumors are regarded as immune-quiescent tumors and non-responsive to single-checkpoint blockade therapies. Many preclinical and clinical studies suggest that it is still important to clarify the regulatory mechanism of the PD-1&#x2F; PD-L1 pathway in pancreatic cancer. As a marker of cancer stem cells, DCLK1 has been found to play an important role in the occurrence and development of a plethora of human cancers. Recent researches have revealed that DCLK1 is closely related to EMT process of tumor cells, meanwhile, it could also be used as a biomarker in gastrointestinal tumors to predict the prognoses of patients. However, the role that DCLK1 plays in the immune regulation of tumor microenvironments remains unknown. Therefore, we sought to understand if DCLK1 could positively regulate the expression of PD-L1 in pancreatic cancer cells. Furthermore, we examined if DCLK1 highly correlated with the Hippo pathway through TCGA database analysis. We found that DCLK1 helped regulate the level of PD-L1 expression by affecting the corresponding expression level of yes-associated protein in the Hippo pathway. Collectively, our study identifies DCLK1 as an important regulator of PD-L1 expression in pancreatic tumor and highlights a central role of DCLK1 in the regulation of tumor immunity. <a href="/publication/inhibition-of-dclk1-down-regulates-pd-l1-expression-through-hippo-pathway-in-human-pancreatic-cancer/" class="post-more waves-effect waves-button">Continue reading...</a></div></article></li></ul><nav id="page-nav"><div class="inner"> <span class="page-number current">1</span><a class="page-number" href="/publication/page/2/">2</a><a class="extend next" rel="next" href="/publication/page/2/">Next</a></div></nav></div><footer class="footer"><div class="top"><p> <span>This page is available under the <a rel="license noopener" target="_blank" href="//creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>.</span></p></div><div class="bottom"><p><span>Li Yao &copy; 2019 - 2025</span> <span>Modified from indigo.</span></p></div></footer></main><div class="mask" id="mask"></div><a href="javascript:;" rel="external nofollow noreferrer" id="gotop" class="waves-effect waves-circle waves-light"><span class="fa-solid fa-lg fa-chevron-up"></span></a><script src="/js/waves.min.js?v=2.0.0"></script><script>MathJax={tex:{inlineMath:[["$","$"],["\\(","\\)"]]},svg:{fontCache:"global"}}</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script><script>var BLOG={ROOT:"/",SHARE:!1,REWARD:!1}</script><script src="/js/main.js"></script><div class="search-panel" id="search-panel"><ul class="search-result" id="search-result"></ul></div><template id="search-tpl"><li class="item"><a href="{path}" class="waves-block waves-effect"><div class="title ellipsis" title="{title}">{title}</div><div class="flex-row flex-middle"><div class="tags ellipsis"> {tags}</div> <time class="flex-col time">{date}</time></div></a></li></template><script src="/js/search.min.js?v=2.0.0" async></script></body></html>